{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "c77f8da1-2837-4bf7-b3f4-dfcaa7ab42f6",
   "metadata": {
    "execution": {
     "iopub.execute_input": "2023-11-08T16:08:10.339493Z",
     "iopub.status.busy": "2023-11-08T16:08:10.339359Z",
     "iopub.status.idle": "2023-11-08T16:08:10.428121Z",
     "shell.execute_reply": "2023-11-08T16:08:10.427511Z"
    }
   },
   "outputs": [],
   "source": [
    "import scipy.io\n",
    "import numpy as np\n",
    "import random\n",
    "import os"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "3fe25376-b856-4296-ba02-5e3771ff9451",
   "metadata": {
    "collapsed": true,
    "execution": {
     "iopub.execute_input": "2023-11-08T16:08:10.430715Z",
     "iopub.status.busy": "2023-11-08T16:08:10.430538Z",
     "iopub.status.idle": "2023-11-08T17:57:45.395645Z",
     "shell.execute_reply": "2023-11-08T17:57:45.395092Z"
    },
    "jupyter": {
     "outputs_hidden": true
    }
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "2023-11-08 11:08:10.615283: I tensorflow/core/platform/cpu_feature_guard.cc:182] This TensorFlow binary is optimized to use available CPU instructions in performance-critical operations.\n",
      "To enable the following instructions: FMA, in other operations, rebuild TensorFlow with the appropriate compiler flags.\n",
      "/usr/lib/python3.11/site-packages/h5py/__init__.py:36: UserWarning: h5py is running against HDF5 1.14.2 when it was built against 1.14.1, this may cause problems\n",
      "  _warn((\"h5py is running against HDF5 {0} when it was built against {1}, \"\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "35033\n",
      "WARNING:tensorflow:From /usr/lib/python3.11/site-packages/tensorflow/python/compat/v2_compat.py:107: disable_resource_variables (from tensorflow.python.ops.variable_scope) is deprecated and will be removed in a future version.\n",
      "Instructions for updating:\n",
      "non-resource variables are not supported in the long term\n",
      "WARNING:tensorflow:Layer lstm will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_1 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_2 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_3 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_4 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "2023-11-08 11:09:36.132722: I tensorflow/compiler/xla/stream_executor/cuda/cuda_gpu_executor.cc:995] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero. See more at https://github.com/torvalds/linux/blob/v6.0/Documentation/ABI/testing/sysfs-bus-pci#L344-L355\n",
      "2023-11-08 11:09:36.144255: I tensorflow/compiler/xla/stream_executor/cuda/cuda_gpu_executor.cc:995] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero. See more at https://github.com/torvalds/linux/blob/v6.0/Documentation/ABI/testing/sysfs-bus-pci#L344-L355\n",
      "2023-11-08 11:09:36.144504: I tensorflow/compiler/xla/stream_executor/cuda/cuda_gpu_executor.cc:995] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero. See more at https://github.com/torvalds/linux/blob/v6.0/Documentation/ABI/testing/sysfs-bus-pci#L344-L355\n",
      "2023-11-08 11:09:36.148114: I tensorflow/compiler/xla/stream_executor/cuda/cuda_gpu_executor.cc:995] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero. See more at https://github.com/torvalds/linux/blob/v6.0/Documentation/ABI/testing/sysfs-bus-pci#L344-L355\n",
      "2023-11-08 11:09:36.148314: I tensorflow/compiler/xla/stream_executor/cuda/cuda_gpu_executor.cc:995] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero. See more at https://github.com/torvalds/linux/blob/v6.0/Documentation/ABI/testing/sysfs-bus-pci#L344-L355\n",
      "2023-11-08 11:09:36.148481: I tensorflow/compiler/xla/stream_executor/cuda/cuda_gpu_executor.cc:995] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero. See more at https://github.com/torvalds/linux/blob/v6.0/Documentation/ABI/testing/sysfs-bus-pci#L344-L355\n",
      "2023-11-08 11:09:36.242830: I tensorflow/compiler/xla/stream_executor/cuda/cuda_gpu_executor.cc:995] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero. See more at https://github.com/torvalds/linux/blob/v6.0/Documentation/ABI/testing/sysfs-bus-pci#L344-L355\n",
      "2023-11-08 11:09:36.243034: I tensorflow/compiler/xla/stream_executor/cuda/cuda_gpu_executor.cc:995] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero. See more at https://github.com/torvalds/linux/blob/v6.0/Documentation/ABI/testing/sysfs-bus-pci#L344-L355\n",
      "2023-11-08 11:09:36.243201: I tensorflow/compiler/xla/stream_executor/cuda/cuda_gpu_executor.cc:995] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero. See more at https://github.com/torvalds/linux/blob/v6.0/Documentation/ABI/testing/sysfs-bus-pci#L344-L355\n",
      "2023-11-08 11:09:36.243338: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1639] Created device /job:localhost/replica:0/task:0/device:GPU:0 with 46608 MB memory:  -> device: 0, name: NVIDIA RTX A6000, pci bus id: 0000:41:00.0, compute capability: 8.6\n",
      "2023-11-08 11:09:36.243721: I tensorflow/core/common_runtime/process_util.cc:146] Creating new thread pool with default inter op setting: 2. Tune using inter_op_parallelism_threads for best performance.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "WARNING:tensorflow:Layer lstm_5 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_6 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_7 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_8 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_9 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_10 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_11 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_12 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_13 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_14 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_15 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_16 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_17 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_18 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_19 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_20 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_21 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "2023-11-08 11:09:37.183506: I tensorflow/compiler/xla/stream_executor/cuda/cuda_gpu_executor.cc:995] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero. See more at https://github.com/torvalds/linux/blob/v6.0/Documentation/ABI/testing/sysfs-bus-pci#L344-L355\n",
      "2023-11-08 11:09:37.183741: I tensorflow/compiler/xla/stream_executor/cuda/cuda_gpu_executor.cc:995] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero. See more at https://github.com/torvalds/linux/blob/v6.0/Documentation/ABI/testing/sysfs-bus-pci#L344-L355\n",
      "2023-11-08 11:09:37.183910: I tensorflow/compiler/xla/stream_executor/cuda/cuda_gpu_executor.cc:995] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero. See more at https://github.com/torvalds/linux/blob/v6.0/Documentation/ABI/testing/sysfs-bus-pci#L344-L355\n",
      "2023-11-08 11:09:37.184173: I tensorflow/compiler/xla/stream_executor/cuda/cuda_gpu_executor.cc:995] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero. See more at https://github.com/torvalds/linux/blob/v6.0/Documentation/ABI/testing/sysfs-bus-pci#L344-L355\n",
      "2023-11-08 11:09:37.184358: I tensorflow/compiler/xla/stream_executor/cuda/cuda_gpu_executor.cc:995] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero. See more at https://github.com/torvalds/linux/blob/v6.0/Documentation/ABI/testing/sysfs-bus-pci#L344-L355\n",
      "2023-11-08 11:09:37.184494: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1639] Created device /job:localhost/replica:0/task:0/device:GPU:0 with 46608 MB memory:  -> device: 0, name: NVIDIA RTX A6000, pci bus id: 0000:41:00.0, compute capability: 8.6\n",
      "2023-11-08 11:09:37.184525: I tensorflow/core/common_runtime/process_util.cc:146] Creating new thread pool with default inter op setting: 2. Tune using inter_op_parallelism_threads for best performance.\n",
      "2023-11-08 11:09:37.206350: I tensorflow/compiler/mlir/mlir_graph_optimization_pass.cc:375] MLIR V1 optimization pass is not enabled\n",
      "2023-11-08 11:09:37.420775: W tensorflow/c/c_api.cc:304] Operation '{name:'lstm_8/lstm_cell_8/bias/Assign' id:1365 op device:{requested: '', assigned: ''} def:{{{node lstm_8/lstm_cell_8/bias/Assign}} = AssignVariableOp[_has_manual_control_dependencies=true, dtype=DT_FLOAT, validate_shape=false](lstm_8/lstm_cell_8/bias, lstm_8/lstm_cell_8/bias/Initializer/concat)}}' was changed by setting attribute after it was run by a session. This mutation will have no effect, and will trigger an error in the future. Either don't modify nodes after running them or create a new session.\n",
      "2023-11-08 11:09:37.556028: W tensorflow/c/c_api.cc:304] Operation '{name:'strided_slice/stack_1' id:3634 op device:{requested: '', assigned: ''} def:{{{node strided_slice/stack_1}} = Const[_has_manual_control_dependencies=true, dtype=DT_INT32, value=Tensor<type: int32 shape: [4] values: 0 0 0...>]()}}' was changed by setting attribute after it was run by a session. This mutation will have no effect, and will trigger an error in the future. Either don't modify nodes after running them or create a new session.\n",
      "2023-11-08 11:09:37.591170: W tensorflow/c/c_api.cc:304] Operation '{name:'strided_slice/stack_2' id:3635 op device:{requested: '', assigned: ''} def:{{{node strided_slice/stack_2}} = Const[_has_manual_control_dependencies=true, dtype=DT_INT32, value=Tensor<type: int32 shape: [4] values: 1 1 1...>]()}}' was changed by setting attribute after it was run by a session. This mutation will have no effect, and will trigger an error in the future. Either don't modify nodes after running them or create a new session.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "WARNING:tensorflow:Layer lstm_22 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_23 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_24 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_25 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_26 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_27 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_28 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_29 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_30 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_31 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_32 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_33 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_34 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_35 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_36 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "(31519, 95)\n",
      "Train on 31519 samples, validate on 3514 samples\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "2023-11-08 11:09:42.075480: W tensorflow/c/c_api.cc:304] Operation '{name:'training/Adam/conv2d_1/bias/v/Assign' id:17012 op device:{requested: '', assigned: ''} def:{{{node training/Adam/conv2d_1/bias/v/Assign}} = AssignVariableOp[_has_manual_control_dependencies=true, dtype=DT_FLOAT, validate_shape=false](training/Adam/conv2d_1/bias/v, training/Adam/conv2d_1/bias/v/Initializer/zeros)}}' was changed by setting attribute after it was run by a session. This mutation will have no effect, and will trigger an error in the future. Either don't modify nodes after running them or create a new session.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Epoch 1/50\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "2023-11-08 11:09:46.419465: E tensorflow/core/grappler/optimizers/meta_optimizer.cc:954] layout failed: INVALID_ARGUMENT: Size of values 0 does not match size of permutation 4 @ fanin shape indropout/cond/dropout/SelectV2-2-TransposeNHWCToNCHW-LayoutOptimizer\n",
      "2023-11-08 11:09:48.849132: I tensorflow/compiler/xla/stream_executor/cuda/cuda_dnn.cc:432] Loaded cuDNN version 8902\n",
      "2023-11-08 11:09:49.066801: I tensorflow/compiler/xla/stream_executor/cuda/cuda_blas.cc:606] TensorFloat-32 will be used for the matrix multiplication. This will only be logged once.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "31519/31519 [==============================] - ETA: 0s - loss: 3.2873"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/usr/lib/python3.11/site-packages/keras/engine/training_v1.py:2335: UserWarning: `Model.state_updates` will be removed in a future version. This property should not be used in TensorFlow 2.0, as `updates` are applied automatically.\n",
      "  updates = self.state_updates\n",
      "2023-11-08 11:10:11.221507: W tensorflow/c/c_api.cc:304] Operation '{name:'loss_1/mul' id:6477 op device:{requested: '', assigned: ''} def:{{{node loss_1/mul}} = Mul[T=DT_FLOAT, _has_manual_control_dependencies=true](loss_1/mul/x, loss_1/dense_3_loss/value)}}' was changed by setting attribute after it was run by a session. This mutation will have no effect, and will trigger an error in the future. Either don't modify nodes after running them or create a new session.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Epoch 1: val_loss improved from inf to 1.93436, saving model to ./checkpoints/unknown_person_baseline_p21_20.h5\n",
      "31519/31519 [==============================] - 30s 955us/sample - loss: 3.2873 - val_loss: 1.9344\n",
      "Epoch 2/50\n",
      "31519/31519 [==============================] - ETA: 0s - loss: 1.8116\n",
      "Epoch 2: val_loss improved from 1.93436 to 1.61149, saving model to ./checkpoints/unknown_person_baseline_p21_20.h5\n",
      "31519/31519 [==============================] - 19s 608us/sample - loss: 1.8116 - val_loss: 1.6115\n",
      "Epoch 3/50\n",
      "31519/31519 [==============================] - ETA: 0s - loss: 1.5956\n",
      "Epoch 3: val_loss improved from 1.61149 to 1.51430, saving model to ./checkpoints/unknown_person_baseline_p21_20.h5\n",
      "31519/31519 [==============================] - 21s 672us/sample - loss: 1.5956 - val_loss: 1.5143\n",
      "Epoch 4/50\n",
      "31519/31519 [==============================] - ETA: 0s - loss: 1.5319\n",
      "Epoch 4: val_loss improved from 1.51430 to 1.48143, saving model to ./checkpoints/unknown_person_baseline_p21_20.h5\n",
      "31519/31519 [==============================] - 22s 700us/sample - loss: 1.5319 - val_loss: 1.4814\n",
      "Epoch 5/50\n",
      "31519/31519 [==============================] - ETA: 0s - loss: 1.5030\n",
      "Epoch 5: val_loss improved from 1.48143 to 1.46594, saving model to ./checkpoints/unknown_person_baseline_p21_20.h5\n",
      "31519/31519 [==============================] - 22s 697us/sample - loss: 1.5030 - val_loss: 1.4659\n",
      "Epoch 6/50\n",
      "31519/31519 [==============================] - ETA: 0s - loss: 1.4823\n",
      "Epoch 6: val_loss improved from 1.46594 to 1.45713, saving model to ./checkpoints/unknown_person_baseline_p21_20.h5\n",
      "31519/31519 [==============================] - 22s 704us/sample - loss: 1.4823 - val_loss: 1.4571\n",
      "Epoch 7/50\n",
      "31519/31519 [==============================] - ETA: 0s - loss: 1.4744\n",
      "Epoch 7: val_loss improved from 1.45713 to 1.44539, saving model to ./checkpoints/unknown_person_baseline_p21_20.h5\n",
      "31519/31519 [==============================] - 22s 701us/sample - loss: 1.4744 - val_loss: 1.4454\n",
      "Epoch 8/50\n",
      "31519/31519 [==============================] - ETA: 0s - loss: 1.4695\n",
      "Epoch 8: val_loss improved from 1.44539 to 1.43578, saving model to ./checkpoints/unknown_person_baseline_p21_20.h5\n",
      "31519/31519 [==============================] - 22s 692us/sample - loss: 1.4695 - val_loss: 1.4358\n",
      "Epoch 9/50\n",
      "31519/31519 [==============================] - ETA: 0s - loss: 1.4649\n",
      "Epoch 9: val_loss improved from 1.43578 to 1.43202, saving model to ./checkpoints/unknown_person_baseline_p21_20.h5\n",
      "31519/31519 [==============================] - 22s 700us/sample - loss: 1.4649 - val_loss: 1.4320\n",
      "Epoch 10/50\n",
      "31519/31519 [==============================] - ETA: 0s - loss: 1.4596\n",
      "Epoch 10: val_loss improved from 1.43202 to 1.42634, saving model to ./checkpoints/unknown_person_baseline_p21_20.h5\n",
      "31519/31519 [==============================] - 22s 689us/sample - loss: 1.4596 - val_loss: 1.4263\n",
      "Epoch 11/50\n",
      "31519/31519 [==============================] - ETA: 0s - loss: 1.4596\n",
      "Epoch 11: val_loss improved from 1.42634 to 1.42510, saving model to ./checkpoints/unknown_person_baseline_p21_20.h5\n",
      "31519/31519 [==============================] - 22s 688us/sample - loss: 1.4596 - val_loss: 1.4251\n",
      "Epoch 12/50\n",
      "31519/31519 [==============================] - ETA: 0s - loss: 1.4686\n",
      "Epoch 12: val_loss did not improve from 1.42510\n",
      "31519/31519 [==============================] - 19s 615us/sample - loss: 1.4686 - val_loss: 1.4267\n",
      "Epoch 13/50\n",
      "31519/31519 [==============================] - ETA: 0s - loss: 1.4726\n",
      "Epoch 13: val_loss did not improve from 1.42510\n",
      "31519/31519 [==============================] - 20s 621us/sample - loss: 1.4726 - val_loss: 1.4277\n",
      "Epoch 14/50\n",
      "31519/31519 [==============================] - ETA: 0s - loss: 1.5110\n",
      "Epoch 14: val_loss did not improve from 1.42510\n",
      "31519/31519 [==============================] - 19s 614us/sample - loss: 1.5110 - val_loss: 1.4356\n",
      "Epoch 15/50\n",
      "31519/31519 [==============================] - ETA: 0s - loss: 1.4715\n",
      "Epoch 15: val_loss improved from 1.42510 to 1.42323, saving model to ./checkpoints/unknown_person_baseline_p21_20.h5\n",
      "31519/31519 [==============================] - 19s 618us/sample - loss: 1.4715 - val_loss: 1.4232\n",
      "Epoch 16/50\n",
      "31519/31519 [==============================] - ETA: 0s - loss: 1.4693\n",
      "Epoch 16: val_loss improved from 1.42323 to 1.42218, saving model to ./checkpoints/unknown_person_baseline_p21_20.h5\n",
      "31519/31519 [==============================] - 19s 617us/sample - loss: 1.4693 - val_loss: 1.4222\n",
      "Epoch 17/50\n",
      "31519/31519 [==============================] - ETA: 0s - loss: 1.5041\n",
      "Epoch 17: val_loss did not improve from 1.42218\n",
      "31519/31519 [==============================] - 19s 614us/sample - loss: 1.5041 - val_loss: 1.4356\n",
      "Epoch 18/50\n",
      "31519/31519 [==============================] - ETA: 0s - loss: 1.4463\n",
      "Epoch 18: val_loss improved from 1.42218 to 1.42127, saving model to ./checkpoints/unknown_person_baseline_p21_20.h5\n",
      "31519/31519 [==============================] - 19s 615us/sample - loss: 1.4463 - val_loss: 1.4213\n",
      "Epoch 19/50\n",
      "31519/31519 [==============================] - ETA: 0s - loss: 1.4594\n",
      "Epoch 19: val_loss did not improve from 1.42127\n",
      "31519/31519 [==============================] - 19s 603us/sample - loss: 1.4594 - val_loss: 1.4242\n",
      "Epoch 20/50\n",
      "31519/31519 [==============================] - ETA: 0s - loss: 1.4758\n",
      "Epoch 20: val_loss improved from 1.42127 to 1.42072, saving model to ./checkpoints/unknown_person_baseline_p21_20.h5\n",
      "31519/31519 [==============================] - 20s 622us/sample - loss: 1.4758 - val_loss: 1.4207\n",
      "Epoch 21/50\n",
      "31519/31519 [==============================] - ETA: 0s - loss: 1.5089\n",
      "Epoch 21: val_loss improved from 1.42072 to 1.41817, saving model to ./checkpoints/unknown_person_baseline_p21_20.h5\n",
      "31519/31519 [==============================] - 21s 656us/sample - loss: 1.5089 - val_loss: 1.4182\n",
      "Epoch 22/50\n",
      "31519/31519 [==============================] - ETA: 0s - loss: 1.4752\n",
      "Epoch 22: val_loss did not improve from 1.41817\n",
      "31519/31519 [==============================] - 20s 645us/sample - loss: 1.4752 - val_loss: 1.4188\n",
      "Epoch 23/50\n",
      "31519/31519 [==============================] - ETA: 0s - loss: 1.4406\n",
      "Epoch 23: val_loss improved from 1.41817 to 1.41644, saving model to ./checkpoints/unknown_person_baseline_p21_20.h5\n",
      "31519/31519 [==============================] - 20s 620us/sample - loss: 1.4406 - val_loss: 1.4164\n",
      "Epoch 24/50\n",
      "31519/31519 [==============================] - ETA: 0s - loss: 1.4287\n",
      "Epoch 24: val_loss improved from 1.41644 to 1.41341, saving model to ./checkpoints/unknown_person_baseline_p21_20.h5\n",
      "31519/31519 [==============================] - 19s 618us/sample - loss: 1.4287 - val_loss: 1.4134\n",
      "Epoch 25/50\n",
      "31519/31519 [==============================] - ETA: 0s - loss: 1.4253\n",
      "Epoch 25: val_loss improved from 1.41341 to 1.40764, saving model to ./checkpoints/unknown_person_baseline_p21_20.h5\n",
      "31519/31519 [==============================] - 19s 611us/sample - loss: 1.4253 - val_loss: 1.4076\n",
      "Epoch 26/50\n",
      "31519/31519 [==============================] - ETA: 0s - loss: 1.4222\n",
      "Epoch 26: val_loss improved from 1.40764 to 1.40644, saving model to ./checkpoints/unknown_person_baseline_p21_20.h5\n",
      "31519/31519 [==============================] - 21s 651us/sample - loss: 1.4222 - val_loss: 1.4064\n",
      "Epoch 27/50\n",
      "31519/31519 [==============================] - ETA: 0s - loss: 1.4200\n",
      "Epoch 27: val_loss did not improve from 1.40644\n",
      "31519/31519 [==============================] - 21s 675us/sample - loss: 1.4200 - val_loss: 1.4070\n",
      "Epoch 28/50\n",
      "31519/31519 [==============================] - ETA: 0s - loss: 1.4145\n",
      "Epoch 28: val_loss improved from 1.40644 to 1.40435, saving model to ./checkpoints/unknown_person_baseline_p21_20.h5\n",
      "31519/31519 [==============================] - 22s 696us/sample - loss: 1.4145 - val_loss: 1.4044\n",
      "Epoch 29/50\n",
      "31519/31519 [==============================] - ETA: 0s - loss: 1.4106\n",
      "Epoch 29: val_loss improved from 1.40435 to 1.39955, saving model to ./checkpoints/unknown_person_baseline_p21_20.h5\n",
      "31519/31519 [==============================] - 21s 662us/sample - loss: 1.4106 - val_loss: 1.3995\n",
      "Epoch 30/50\n",
      "31519/31519 [==============================] - ETA: 0s - loss: 1.4089\n",
      "Epoch 30: val_loss did not improve from 1.39955\n",
      "31519/31519 [==============================] - 21s 681us/sample - loss: 1.4089 - val_loss: 1.4005\n",
      "Epoch 31/50\n",
      "31519/31519 [==============================] - ETA: 0s - loss: 1.4070\n",
      "Epoch 31: val_loss did not improve from 1.39955\n",
      "31519/31519 [==============================] - 20s 643us/sample - loss: 1.4070 - val_loss: 1.4022\n",
      "Epoch 32/50\n",
      "31519/31519 [==============================] - ETA: 0s - loss: 1.4033\n",
      "Epoch 32: val_loss improved from 1.39955 to 1.39547, saving model to ./checkpoints/unknown_person_baseline_p21_20.h5\n",
      "31519/31519 [==============================] - 20s 648us/sample - loss: 1.4033 - val_loss: 1.3955\n",
      "Epoch 33/50\n",
      "31519/31519 [==============================] - ETA: 0s - loss: 1.4046\n",
      "Epoch 33: val_loss did not improve from 1.39547\n",
      "31519/31519 [==============================] - 22s 694us/sample - loss: 1.4046 - val_loss: 1.3988\n",
      "Epoch 34/50\n",
      "31519/31519 [==============================] - ETA: 0s - loss: 1.4015\n",
      "Epoch 34: val_loss improved from 1.39547 to 1.39332, saving model to ./checkpoints/unknown_person_baseline_p21_20.h5\n",
      "31519/31519 [==============================] - 22s 701us/sample - loss: 1.4015 - val_loss: 1.3933\n",
      "Epoch 35/50\n",
      "31519/31519 [==============================] - ETA: 0s - loss: 1.3970\n",
      "Epoch 35: val_loss did not improve from 1.39332\n",
      "31519/31519 [==============================] - 21s 662us/sample - loss: 1.3970 - val_loss: 1.3946\n",
      "Epoch 36/50\n",
      "31519/31519 [==============================] - ETA: 0s - loss: 1.3952\n",
      "Epoch 36: val_loss did not improve from 1.39332\n",
      "31519/31519 [==============================] - 19s 600us/sample - loss: 1.3952 - val_loss: 1.3950\n",
      "Epoch 37/50\n",
      "31519/31519 [==============================] - ETA: 0s - loss: 1.3938\n",
      "Epoch 37: val_loss improved from 1.39332 to 1.39061, saving model to ./checkpoints/unknown_person_baseline_p21_20.h5\n",
      "31519/31519 [==============================] - 19s 599us/sample - loss: 1.3938 - val_loss: 1.3906\n",
      "Epoch 38/50\n",
      "31519/31519 [==============================] - ETA: 0s - loss: 1.3927\n",
      "Epoch 38: val_loss did not improve from 1.39061\n",
      "31519/31519 [==============================] - 20s 645us/sample - loss: 1.3927 - val_loss: 1.3918\n",
      "Epoch 39/50\n",
      "31519/31519 [==============================] - ETA: 0s - loss: 1.3890\n",
      "Epoch 39: val_loss did not improve from 1.39061\n",
      "31519/31519 [==============================] - 22s 702us/sample - loss: 1.3890 - val_loss: 1.3923\n",
      "Epoch 40/50\n",
      "31519/31519 [==============================] - ETA: 0s - loss: 1.3882\n",
      "Epoch 40: val_loss improved from 1.39061 to 1.38881, saving model to ./checkpoints/unknown_person_baseline_p21_20.h5\n",
      "31519/31519 [==============================] - 22s 706us/sample - loss: 1.3882 - val_loss: 1.3888\n",
      "Epoch 41/50\n",
      "31519/31519 [==============================] - ETA: 0s - loss: 1.3855\n",
      "Epoch 41: val_loss improved from 1.38881 to 1.38724, saving model to ./checkpoints/unknown_person_baseline_p21_20.h5\n",
      "31519/31519 [==============================] - 22s 700us/sample - loss: 1.3855 - val_loss: 1.3872\n",
      "Epoch 42/50\n",
      "31519/31519 [==============================] - ETA: 0s - loss: 1.3868\n",
      "Epoch 42: val_loss improved from 1.38724 to 1.38396, saving model to ./checkpoints/unknown_person_baseline_p21_20.h5\n",
      "31519/31519 [==============================] - 22s 690us/sample - loss: 1.3868 - val_loss: 1.3840\n",
      "Epoch 43/50\n",
      "31519/31519 [==============================] - ETA: 0s - loss: 1.3834\n",
      "Epoch 43: val_loss did not improve from 1.38396\n",
      "31519/31519 [==============================] - 19s 618us/sample - loss: 1.3834 - val_loss: 1.3883\n",
      "Epoch 44/50\n",
      "31519/31519 [==============================] - ETA: 0s - loss: 1.3847\n",
      "Epoch 44: val_loss did not improve from 1.38396\n",
      "31519/31519 [==============================] - 20s 619us/sample - loss: 1.3847 - val_loss: 1.3862\n",
      "Epoch 45/50\n",
      "31519/31519 [==============================] - ETA: 0s - loss: 1.3821\n",
      "Epoch 45: val_loss improved from 1.38396 to 1.38391, saving model to ./checkpoints/unknown_person_baseline_p21_20.h5\n",
      "31519/31519 [==============================] - 20s 622us/sample - loss: 1.3821 - val_loss: 1.3839\n",
      "Epoch 46/50\n",
      "31519/31519 [==============================] - ETA: 0s - loss: 1.3799\n",
      "Epoch 46: val_loss improved from 1.38391 to 1.38386, saving model to ./checkpoints/unknown_person_baseline_p21_20.h5\n",
      "31519/31519 [==============================] - 21s 652us/sample - loss: 1.3799 - val_loss: 1.3839\n",
      "Epoch 47/50\n",
      "31519/31519 [==============================] - ETA: 0s - loss: 1.3800\n",
      "Epoch 47: val_loss improved from 1.38386 to 1.38158, saving model to ./checkpoints/unknown_person_baseline_p21_20.h5\n",
      "31519/31519 [==============================] - 21s 655us/sample - loss: 1.3800 - val_loss: 1.3816\n",
      "Epoch 48/50\n",
      "31519/31519 [==============================] - ETA: 0s - loss: 1.3782\n",
      "Epoch 48: val_loss improved from 1.38158 to 1.37819, saving model to ./checkpoints/unknown_person_baseline_p21_20.h5\n",
      "31519/31519 [==============================] - 22s 702us/sample - loss: 1.3782 - val_loss: 1.3782\n",
      "Epoch 49/50\n",
      "31519/31519 [==============================] - ETA: 0s - loss: 1.3778\n",
      "Epoch 49: val_loss did not improve from 1.37819\n",
      "31519/31519 [==============================] - 22s 699us/sample - loss: 1.3778 - val_loss: 1.3803\n",
      "Epoch 50/50\n",
      "31519/31519 [==============================] - ETA: 0s - loss: 1.3784\n",
      "Epoch 50: val_loss did not improve from 1.37819\n",
      "31519/31519 [==============================] - 20s 645us/sample - loss: 1.3784 - val_loss: 1.3830\n",
      "WARNING:tensorflow:From /usr/lib/python3.11/site-packages/tensorflow/python/ops/init_ops.py:94: calling GlorotUniform.__init__ (from tensorflow.python.ops.init_ops) with dtype is deprecated and will be removed in a future version.\n",
      "Instructions for updating:\n",
      "Call initializer instance with the dtype argument instead of passing it to the constructor\n",
      "WARNING:tensorflow:From /usr/lib/python3.11/site-packages/tensorflow/python/ops/init_ops.py:94: calling Zeros.__init__ (from tensorflow.python.ops.init_ops) with dtype is deprecated and will be removed in a future version.\n",
      "Instructions for updating:\n",
      "Call initializer instance with the dtype argument instead of passing it to the constructor\n",
      "WARNING:tensorflow:From /usr/lib/python3.11/site-packages/tensorflow/python/ops/init_ops.py:94: calling Orthogonal.__init__ (from tensorflow.python.ops.init_ops) with dtype is deprecated and will be removed in a future version.\n",
      "Instructions for updating:\n",
      "Call initializer instance with the dtype argument instead of passing it to the constructor\n",
      "WARNING:tensorflow:Layer lstm will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_1 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_2 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_3 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_4 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_5 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_6 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_7 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_8 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_9 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_10 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_11 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_12 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_13 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_14 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_15 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_16 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_17 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_18 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_19 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_20 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_21 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_22 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_23 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_24 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_25 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_26 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_27 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_28 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_29 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_30 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_31 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_32 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_33 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_34 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_35 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_36 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "2023-11-08 11:27:09.802088: W tensorflow/c/c_api.cc:304] Operation '{name:'lstm_13_1/lstm_cell_50/kernel/Assign' id:20720 op device:{requested: '', assigned: ''} def:{{{node lstm_13_1/lstm_cell_50/kernel/Assign}} = AssignVariableOp[_has_manual_control_dependencies=true, dtype=DT_FLOAT, validate_shape=false](lstm_13_1/lstm_cell_50/kernel, lstm_13_1/lstm_cell_50/kernel/Initializer/random_uniform)}}' was changed by setting attribute after it was run by a session. This mutation will have no effect, and will trigger an error in the future. Either don't modify nodes after running them or create a new session.\n",
      "2023-11-08 11:27:11.541338: W tensorflow/c/c_api.cc:304] Operation '{name:'lstm_12_1/lstm_cell_49/kernel/v/Assign' id:26075 op device:{requested: '', assigned: ''} def:{{{node lstm_12_1/lstm_cell_49/kernel/v/Assign}} = AssignVariableOp[_has_manual_control_dependencies=true, dtype=DT_FLOAT, validate_shape=false](lstm_12_1/lstm_cell_49/kernel/v, lstm_12_1/lstm_cell_49/kernel/v/Initializer/zeros)}}' was changed by setting attribute after it was run by a session. This mutation will have no effect, and will trigger an error in the future. Either don't modify nodes after running them or create a new session.\n",
      "/usr/lib/python3.11/site-packages/keras/engine/training_v1.py:2359: UserWarning: `Model.state_updates` will be removed in a future version. This property should not be used in TensorFlow 2.0, as `updates` are applied automatically.\n",
      "  updates=self.state_updates,\n",
      "2023-11-08 11:27:12.934190: W tensorflow/c/c_api.cc:304] Operation '{name:'dropout_4_1/cond/Merge' id:24591 op device:{requested: '', assigned: ''} def:{{{node dropout_4_1/cond/Merge}} = Merge[N=2, T=DT_FLOAT, _has_manual_control_dependencies=true](dropout_4_1/cond/Identity, dropout_4_1/cond/dropout/SelectV2)}}' was changed by setting attribute after it was run by a session. This mutation will have no effect, and will trigger an error in the future. Either don't modify nodes after running them or create a new session.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(1485, 1968)\n",
      "(1514, 1968)\n",
      "(1644, 1968)\n",
      "(1764, 1968)\n",
      "(1836, 1968)\n",
      "(1699, 1968)\n",
      "(1369, 1968)\n",
      "(1766, 1968)\n",
      "(1619, 1968)\n",
      "(1692, 1968)\n",
      "(1550, 1968)\n",
      "(1920, 1968)\n",
      "(1739, 1968)\n",
      "(1872, 1968)\n",
      "(1692, 1968)\n",
      "(1824, 1968)\n",
      "(970, 1968)\n",
      "(1668, 1968)\n",
      "(1896, 1968)\n",
      "{1: 8.552218773156472, 2: 7.290586845760978, 4: 9.160828641680915, 5: 6.229652451762425, 6: 5.3948448543040755, 8: 10.0, 9: 7.219309720622522, 10: 8.661938418387315, 11: 7.254255273705251, 12: 9.723979357727666, 13: 7.327178030377665, 17: 9.79096932899059, 19: 7.949204051847482, 22: 1.0, 25: 7.552243678308039, 26: 7.198164774059046, 27: 5.156642527059526, 28: 5.890503397740472, 29: 2.3118832723918987}\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/tmp/ipykernel_2791918/231707908.py:300: DeprecationWarning: Conversion of an array with ndim > 0 to a scalar is deprecated, and will error in future. Ensure you extract a single element from your array before performing this operation. (Deprecated NumPy 1.25.)\n",
      "  w_train[i] = weights_dict[int(p_train[i])]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Train on 31519 samples, validate on 3514 samples\n",
      "Epoch 1/20\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "2023-11-08 11:31:16.487448: E tensorflow/core/grappler/optimizers/meta_optimizer.cc:954] layout failed: INVALID_ARGUMENT: Size of values 0 does not match size of permutation 4 @ fanin shape indropout_5/cond/dropout/SelectV2-2-TransposeNHWCToNCHW-LayoutOptimizer\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "31519/31519 [==============================] - ETA: 0s - loss: 10.1105\n",
      "Epoch 1: val_loss improved from inf to 1.38776, saving model to ./checkpoints/unknown_person_p21_20.h5\n",
      "31519/31519 [==============================] - 28s 894us/sample - loss: 10.1105 - val_loss: 1.3878\n",
      "Epoch 2/20\n",
      "31519/31519 [==============================] - ETA: 0s - loss: 9.9733\n",
      "Epoch 2: val_loss did not improve from 1.38776\n",
      "31519/31519 [==============================] - 22s 695us/sample - loss: 9.9733 - val_loss: 1.3900\n",
      "Epoch 3/20\n",
      "31519/31519 [==============================] - ETA: 0s - loss: 9.9920 \n",
      "Epoch 3: val_loss did not improve from 1.38776\n",
      "31519/31519 [==============================] - 22s 696us/sample - loss: 9.9920 - val_loss: 1.4025\n",
      "Epoch 4/20\n",
      "31519/31519 [==============================] - ETA: 0s - loss: 9.9373\n",
      "Epoch 4: val_loss improved from 1.38776 to 1.38344, saving model to ./checkpoints/unknown_person_p21_20.h5\n",
      "31519/31519 [==============================] - 20s 620us/sample - loss: 9.9373 - val_loss: 1.3834\n",
      "Epoch 5/20\n",
      "31519/31519 [==============================] - ETA: 0s - loss: 9.8793\n",
      "Epoch 5: val_loss improved from 1.38344 to 1.37345, saving model to ./checkpoints/unknown_person_p21_20.h5\n",
      "31519/31519 [==============================] - 19s 612us/sample - loss: 9.8793 - val_loss: 1.3735\n",
      "Epoch 6/20\n",
      "31519/31519 [==============================] - ETA: 0s - loss: 9.8832\n",
      "Epoch 6: val_loss did not improve from 1.37345\n",
      "31519/31519 [==============================] - 21s 658us/sample - loss: 9.8832 - val_loss: 1.3869\n",
      "Epoch 7/20\n",
      "31519/31519 [==============================] - ETA: 0s - loss: 9.8643\n",
      "Epoch 7: val_loss did not improve from 1.37345\n",
      "31519/31519 [==============================] - 21s 656us/sample - loss: 9.8643 - val_loss: 1.3842\n",
      "Epoch 8/20\n",
      "31519/31519 [==============================] - ETA: 0s - loss: 9.8349\n",
      "Epoch 8: val_loss did not improve from 1.37345\n",
      "31519/31519 [==============================] - 19s 617us/sample - loss: 9.8349 - val_loss: 1.3749\n",
      "Epoch 9/20\n",
      "31519/31519 [==============================] - ETA: 0s - loss: 9.8088\n",
      "Epoch 9: val_loss improved from 1.37345 to 1.37311, saving model to ./checkpoints/unknown_person_p21_20.h5\n",
      "31519/31519 [==============================] - 21s 677us/sample - loss: 9.8088 - val_loss: 1.3731\n",
      "Epoch 10/20\n",
      "31519/31519 [==============================] - ETA: 0s - loss: 9.8013\n",
      "Epoch 10: val_loss did not improve from 1.37311\n",
      "31519/31519 [==============================] - 21s 682us/sample - loss: 9.8013 - val_loss: 1.3953\n",
      "Epoch 11/20\n",
      "31519/31519 [==============================] - ETA: 0s - loss: 9.7928\n",
      "Epoch 11: val_loss did not improve from 1.37311\n",
      "31519/31519 [==============================] - 21s 658us/sample - loss: 9.7928 - val_loss: 1.3785\n",
      "Epoch 12/20\n",
      "31519/31519 [==============================] - ETA: 0s - loss: 9.7814\n",
      "Epoch 12: val_loss did not improve from 1.37311\n",
      "31519/31519 [==============================] - 20s 634us/sample - loss: 9.7814 - val_loss: 1.3734\n",
      "Epoch 13/20\n",
      "31519/31519 [==============================] - ETA: 0s - loss: 9.7608\n",
      "Epoch 13: val_loss improved from 1.37311 to 1.37087, saving model to ./checkpoints/unknown_person_p21_20.h5\n",
      "31519/31519 [==============================] - 20s 644us/sample - loss: 9.7608 - val_loss: 1.3709\n",
      "Epoch 14/20\n",
      "31519/31519 [==============================] - ETA: 0s - loss: 9.7443\n",
      "Epoch 14: val_loss did not improve from 1.37087\n",
      "31519/31519 [==============================] - 20s 623us/sample - loss: 9.7443 - val_loss: 1.3715\n",
      "Epoch 15/20\n",
      "31519/31519 [==============================] - ETA: 0s - loss: 9.7493\n",
      "Epoch 15: val_loss improved from 1.37087 to 1.36571, saving model to ./checkpoints/unknown_person_p21_20.h5\n",
      "31519/31519 [==============================] - 19s 618us/sample - loss: 9.7493 - val_loss: 1.3657\n",
      "Epoch 16/20\n",
      "31519/31519 [==============================] - ETA: 0s - loss: 9.7149\n",
      "Epoch 16: val_loss did not improve from 1.36571\n",
      "31519/31519 [==============================] - 20s 628us/sample - loss: 9.7149 - val_loss: 1.3737\n",
      "Epoch 17/20\n",
      "31519/31519 [==============================] - ETA: 0s - loss: 9.6892\n",
      "Epoch 17: val_loss did not improve from 1.36571\n",
      "31519/31519 [==============================] - 20s 636us/sample - loss: 9.6892 - val_loss: 1.3707\n",
      "Epoch 18/20\n",
      "31519/31519 [==============================] - ETA: 0s - loss: 9.7211\n",
      "Epoch 18: val_loss did not improve from 1.36571\n",
      "31519/31519 [==============================] - 20s 631us/sample - loss: 9.7211 - val_loss: 1.3740\n",
      "Epoch 19/20\n",
      "31519/31519 [==============================] - ETA: 0s - loss: 9.6785\n",
      "Epoch 19: val_loss did not improve from 1.36571\n",
      "31519/31519 [==============================] - 20s 624us/sample - loss: 9.6785 - val_loss: 1.3731\n",
      "Epoch 20/20\n",
      "31519/31519 [==============================] - ETA: 0s - loss: 9.6669\n",
      "Epoch 20: val_loss did not improve from 1.36571\n",
      "31519/31519 [==============================] - 19s 617us/sample - loss: 9.6669 - val_loss: 1.3662\n",
      "WARNING:tensorflow:Layer lstm will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_1 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_2 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_3 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_4 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_5 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_6 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_7 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_8 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_9 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_10 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_11 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_12 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_13 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_14 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_15 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_16 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_17 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_18 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_19 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_20 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_21 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_22 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_23 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_24 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_25 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_26 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_27 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_28 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_29 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_30 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_31 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_32 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_33 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_34 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_35 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_36 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "2023-11-08 11:38:11.798718: W tensorflow/c/c_api.cc:304] Operation '{name:'lstm_12_2/lstm_cell_86/bias/Assign' id:39987 op device:{requested: '', assigned: ''} def:{{{node lstm_12_2/lstm_cell_86/bias/Assign}} = AssignVariableOp[_has_manual_control_dependencies=true, dtype=DT_FLOAT, validate_shape=false](lstm_12_2/lstm_cell_86/bias, lstm_12_2/lstm_cell_86/bias/Initializer/concat)}}' was changed by setting attribute after it was run by a session. This mutation will have no effect, and will trigger an error in the future. Either don't modify nodes after running them or create a new session.\n",
      "2023-11-08 11:38:14.231421: W tensorflow/c/c_api.cc:304] Operation '{name:'lstm_5_2/lstm_cell_79/bias/m/Assign' id:44735 op device:{requested: '', assigned: ''} def:{{{node lstm_5_2/lstm_cell_79/bias/m/Assign}} = AssignVariableOp[_has_manual_control_dependencies=true, dtype=DT_FLOAT, validate_shape=false](lstm_5_2/lstm_cell_79/bias/m, lstm_5_2/lstm_cell_79/bias/m/Initializer/zeros)}}' was changed by setting attribute after it was run by a session. This mutation will have no effect, and will trigger an error in the future. Either don't modify nodes after running them or create a new session.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Train on 31519 samples, validate on 3514 samples\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "2023-11-08 11:38:19.359430: W tensorflow/c/c_api.cc:304] Operation '{name:'loss_5/mul' id:44579 op device:{requested: '', assigned: ''} def:{{{node loss_5/mul}} = Mul[T=DT_FLOAT, _has_manual_control_dependencies=true](loss_5/mul/x, loss_5/dense_3_loss/value)}}' was changed by setting attribute after it was run by a session. This mutation will have no effect, and will trigger an error in the future. Either don't modify nodes after running them or create a new session.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Epoch 1/20\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "2023-11-08 11:38:31.878839: E tensorflow/core/grappler/optimizers/meta_optimizer.cc:954] layout failed: INVALID_ARGUMENT: Size of values 0 does not match size of permutation 4 @ fanin shape indropout_6/cond/dropout/SelectV2-2-TransposeNHWCToNCHW-LayoutOptimizer\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "31519/31519 [==============================] - ETA: 0s - loss: 1.3748"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "2023-11-08 11:38:52.679975: W tensorflow/c/c_api.cc:304] Operation '{name:'loss_5/mul' id:44579 op device:{requested: '', assigned: ''} def:{{{node loss_5/mul}} = Mul[T=DT_FLOAT, _has_manual_control_dependencies=true](loss_5/mul/x, loss_5/dense_3_loss/value)}}' was changed by setting attribute after it was run by a session. This mutation will have no effect, and will trigger an error in the future. Either don't modify nodes after running them or create a new session.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Epoch 1: val_loss improved from inf to 1.37975, saving model to ./checkpoints/unknown_person_baseline_2_p21_20.h5\n",
      "31519/31519 [==============================] - 27s 862us/sample - loss: 1.3748 - val_loss: 1.3797\n",
      "Epoch 2/20\n",
      "31519/31519 [==============================] - ETA: 0s - loss: 1.3781\n",
      "Epoch 2: val_loss did not improve from 1.37975\n",
      "31519/31519 [==============================] - 20s 622us/sample - loss: 1.3781 - val_loss: 1.3812\n",
      "Epoch 3/20\n",
      "31519/31519 [==============================] - ETA: 0s - loss: 1.3744\n",
      "Epoch 3: val_loss improved from 1.37975 to 1.37920, saving model to ./checkpoints/unknown_person_baseline_2_p21_20.h5\n",
      "31519/31519 [==============================] - 20s 626us/sample - loss: 1.3744 - val_loss: 1.3792\n",
      "Epoch 4/20\n",
      "31519/31519 [==============================] - ETA: 0s - loss: 1.3742\n",
      "Epoch 4: val_loss did not improve from 1.37920\n",
      "31519/31519 [==============================] - 20s 625us/sample - loss: 1.3742 - val_loss: 1.3828\n",
      "Epoch 5/20\n",
      "31519/31519 [==============================] - ETA: 0s - loss: 1.3697\n",
      "Epoch 5: val_loss improved from 1.37920 to 1.37640, saving model to ./checkpoints/unknown_person_baseline_2_p21_20.h5\n",
      "31519/31519 [==============================] - 20s 625us/sample - loss: 1.3697 - val_loss: 1.3764\n",
      "Epoch 6/20\n",
      "31519/31519 [==============================] - ETA: 0s - loss: 1.3725\n",
      "Epoch 6: val_loss did not improve from 1.37640\n",
      "31519/31519 [==============================] - 20s 624us/sample - loss: 1.3725 - val_loss: 1.3766\n",
      "Epoch 7/20\n",
      "31519/31519 [==============================] - ETA: 0s - loss: 1.3688\n",
      "Epoch 7: val_loss did not improve from 1.37640\n",
      "31519/31519 [==============================] - 20s 630us/sample - loss: 1.3688 - val_loss: 1.3780\n",
      "Epoch 8/20\n",
      "31519/31519 [==============================] - ETA: 0s - loss: 1.3685\n",
      "Epoch 8: val_loss did not improve from 1.37640\n",
      "31519/31519 [==============================] - 20s 622us/sample - loss: 1.3685 - val_loss: 1.3822\n",
      "Epoch 9/20\n",
      "31519/31519 [==============================] - ETA: 0s - loss: 1.3663\n",
      "Epoch 9: val_loss improved from 1.37640 to 1.37371, saving model to ./checkpoints/unknown_person_baseline_2_p21_20.h5\n",
      "31519/31519 [==============================] - 22s 683us/sample - loss: 1.3663 - val_loss: 1.3737\n",
      "Epoch 10/20\n",
      "31519/31519 [==============================] - ETA: 0s - loss: 1.3641\n",
      "Epoch 10: val_loss improved from 1.37371 to 1.37354, saving model to ./checkpoints/unknown_person_baseline_2_p21_20.h5\n",
      "31519/31519 [==============================] - 22s 692us/sample - loss: 1.3641 - val_loss: 1.3735\n",
      "Epoch 11/20\n",
      "31519/31519 [==============================] - ETA: 0s - loss: 1.3658\n",
      "Epoch 11: val_loss improved from 1.37354 to 1.37051, saving model to ./checkpoints/unknown_person_baseline_2_p21_20.h5\n",
      "31519/31519 [==============================] - 22s 706us/sample - loss: 1.3658 - val_loss: 1.3705\n",
      "Epoch 12/20\n",
      "31519/31519 [==============================] - ETA: 0s - loss: 1.3644\n",
      "Epoch 12: val_loss did not improve from 1.37051\n",
      "31519/31519 [==============================] - 22s 702us/sample - loss: 1.3644 - val_loss: 1.3760\n",
      "Epoch 13/20\n",
      "31519/31519 [==============================] - ETA: 0s - loss: 1.3602\n",
      "Epoch 13: val_loss improved from 1.37051 to 1.36916, saving model to ./checkpoints/unknown_person_baseline_2_p21_20.h5\n",
      "31519/31519 [==============================] - 22s 702us/sample - loss: 1.3602 - val_loss: 1.3692\n",
      "Epoch 14/20\n",
      "31519/31519 [==============================] - ETA: 0s - loss: 1.3616\n",
      "Epoch 14: val_loss did not improve from 1.36916\n",
      "31519/31519 [==============================] - 22s 702us/sample - loss: 1.3616 - val_loss: 1.3728\n",
      "Epoch 15/20\n",
      "31519/31519 [==============================] - ETA: 0s - loss: 1.3580\n",
      "Epoch 15: val_loss did not improve from 1.36916\n",
      "31519/31519 [==============================] - 21s 655us/sample - loss: 1.3580 - val_loss: 1.3715\n",
      "Epoch 16/20\n",
      "31519/31519 [==============================] - ETA: 0s - loss: 1.3596\n",
      "Epoch 16: val_loss improved from 1.36916 to 1.36573, saving model to ./checkpoints/unknown_person_baseline_2_p21_20.h5\n",
      "31519/31519 [==============================] - 22s 700us/sample - loss: 1.3596 - val_loss: 1.3657\n",
      "Epoch 17/20\n",
      "31519/31519 [==============================] - ETA: 0s - loss: 1.3546\n",
      "Epoch 17: val_loss did not improve from 1.36573\n",
      "31519/31519 [==============================] - 22s 698us/sample - loss: 1.3546 - val_loss: 1.3717\n",
      "Epoch 18/20\n",
      "31519/31519 [==============================] - ETA: 0s - loss: 1.3550\n",
      "Epoch 18: val_loss did not improve from 1.36573\n",
      "31519/31519 [==============================] - 22s 698us/sample - loss: 1.3550 - val_loss: 1.3719\n",
      "Epoch 19/20\n",
      "31519/31519 [==============================] - ETA: 0s - loss: 1.3551\n",
      "Epoch 19: val_loss improved from 1.36573 to 1.36505, saving model to ./checkpoints/unknown_person_baseline_2_p21_20.h5\n",
      "31519/31519 [==============================] - 21s 682us/sample - loss: 1.3551 - val_loss: 1.3651\n",
      "Epoch 20/20\n",
      "31519/31519 [==============================] - ETA: 0s - loss: 1.3505\n",
      "Epoch 20: val_loss did not improve from 1.36505\n",
      "31519/31519 [==============================] - 20s 639us/sample - loss: 1.3505 - val_loss: 1.3659\n",
      "Train on 31519 samples, validate on 3514 samples\n",
      "Epoch 1/50\n",
      "31519/31519 [==============================] - ETA: 0s - loss: 1.3543\n",
      "Epoch 1: val_loss improved from inf to 1.36352, saving model to ./checkpoints/unknown_person_baseline_p21_21.h5\n",
      "31519/31519 [==============================] - 20s 624us/sample - loss: 1.3543 - val_loss: 1.3635\n",
      "Epoch 2/50\n",
      "31519/31519 [==============================] - ETA: 0s - loss: 1.3503\n",
      "Epoch 2: val_loss did not improve from 1.36352\n",
      "31519/31519 [==============================] - 20s 628us/sample - loss: 1.3503 - val_loss: 1.3658\n",
      "Epoch 3/50\n",
      "31519/31519 [==============================] - ETA: 0s - loss: 1.3493\n",
      "Epoch 3: val_loss did not improve from 1.36352\n",
      "31519/31519 [==============================] - 20s 631us/sample - loss: 1.3493 - val_loss: 1.3638\n",
      "Epoch 4/50\n",
      "31519/31519 [==============================] - ETA: 0s - loss: 1.3479\n",
      "Epoch 4: val_loss improved from 1.36352 to 1.36285, saving model to ./checkpoints/unknown_person_baseline_p21_21.h5\n",
      "31519/31519 [==============================] - 20s 636us/sample - loss: 1.3479 - val_loss: 1.3629\n",
      "Epoch 5/50\n",
      "31519/31519 [==============================] - ETA: 0s - loss: 1.3470\n",
      "Epoch 5: val_loss improved from 1.36285 to 1.36065, saving model to ./checkpoints/unknown_person_baseline_p21_21.h5\n",
      "31519/31519 [==============================] - 20s 631us/sample - loss: 1.3470 - val_loss: 1.3606\n",
      "Epoch 6/50\n",
      "31519/31519 [==============================] - ETA: 0s - loss: 1.3481\n",
      "Epoch 6: val_loss did not improve from 1.36065\n",
      "31519/31519 [==============================] - 21s 651us/sample - loss: 1.3481 - val_loss: 1.3643\n",
      "Epoch 7/50\n",
      "31519/31519 [==============================] - ETA: 0s - loss: 1.3454\n",
      "Epoch 7: val_loss did not improve from 1.36065\n",
      "31519/31519 [==============================] - 21s 664us/sample - loss: 1.3454 - val_loss: 1.3643\n",
      "Epoch 8/50\n",
      "31519/31519 [==============================] - ETA: 0s - loss: 1.3494\n",
      "Epoch 8: val_loss did not improve from 1.36065\n",
      "31519/31519 [==============================] - 20s 629us/sample - loss: 1.3494 - val_loss: 1.3608\n",
      "Epoch 9/50\n",
      "31519/31519 [==============================] - ETA: 0s - loss: 1.3534\n",
      "Epoch 9: val_loss did not improve from 1.36065\n",
      "31519/31519 [==============================] - 20s 627us/sample - loss: 1.3534 - val_loss: 1.3671\n",
      "Epoch 10/50\n",
      "31519/31519 [==============================] - ETA: 0s - loss: 1.3513\n",
      "Epoch 10: val_loss did not improve from 1.36065\n",
      "31519/31519 [==============================] - 20s 636us/sample - loss: 1.3513 - val_loss: 1.3686\n",
      "Epoch 11/50\n",
      "31519/31519 [==============================] - ETA: 0s - loss: 1.3629\n",
      "Epoch 11: val_loss did not improve from 1.36065\n",
      "31519/31519 [==============================] - 22s 701us/sample - loss: 1.3629 - val_loss: 1.3654\n",
      "Epoch 12/50\n",
      "31519/31519 [==============================] - ETA: 0s - loss: 1.3629\n",
      "Epoch 12: val_loss did not improve from 1.36065\n",
      "31519/31519 [==============================] - 22s 697us/sample - loss: 1.3629 - val_loss: 1.3639\n",
      "Epoch 13/50\n",
      "31519/31519 [==============================] - ETA: 0s - loss: 1.3603\n",
      "Epoch 13: val_loss did not improve from 1.36065\n",
      "31519/31519 [==============================] - 22s 700us/sample - loss: 1.3603 - val_loss: 1.3683\n",
      "Epoch 14/50\n",
      "31519/31519 [==============================] - ETA: 0s - loss: 1.3687\n",
      "Epoch 14: val_loss improved from 1.36065 to 1.36033, saving model to ./checkpoints/unknown_person_baseline_p21_21.h5\n",
      "31519/31519 [==============================] - 22s 708us/sample - loss: 1.3687 - val_loss: 1.3603\n",
      "Epoch 15/50\n",
      "31519/31519 [==============================] - ETA: 0s - loss: 1.3639\n",
      "Epoch 15: val_loss did not improve from 1.36033\n",
      "31519/31519 [==============================] - 22s 693us/sample - loss: 1.3639 - val_loss: 1.3697\n",
      "Epoch 16/50\n",
      "31519/31519 [==============================] - ETA: 0s - loss: 1.3650\n",
      "Epoch 16: val_loss did not improve from 1.36033\n",
      "31519/31519 [==============================] - 20s 626us/sample - loss: 1.3650 - val_loss: 1.3659\n",
      "Epoch 17/50\n",
      "31519/31519 [==============================] - ETA: 0s - loss: 1.3650\n",
      "Epoch 17: val_loss did not improve from 1.36033\n",
      "31519/31519 [==============================] - 20s 621us/sample - loss: 1.3650 - val_loss: 1.3651\n",
      "Epoch 18/50\n",
      "31519/31519 [==============================] - ETA: 0s - loss: 1.4068\n",
      "Epoch 18: val_loss did not improve from 1.36033\n",
      "31519/31519 [==============================] - 19s 614us/sample - loss: 1.4068 - val_loss: 1.3661\n",
      "Epoch 19/50\n",
      "31519/31519 [==============================] - ETA: 0s - loss: 1.4119\n",
      "Epoch 19: val_loss did not improve from 1.36033\n",
      "31519/31519 [==============================] - 21s 678us/sample - loss: 1.4119 - val_loss: 1.3755\n",
      "Epoch 20/50\n",
      "31519/31519 [==============================] - ETA: 0s - loss: 1.4600\n",
      "Epoch 20: val_loss did not improve from 1.36033\n",
      "31519/31519 [==============================] - 20s 621us/sample - loss: 1.4600 - val_loss: 1.3827\n",
      "Epoch 21/50\n",
      "31519/31519 [==============================] - ETA: 0s - loss: 1.4078\n",
      "Epoch 21: val_loss did not improve from 1.36033\n",
      "31519/31519 [==============================] - 19s 610us/sample - loss: 1.4078 - val_loss: 1.3799\n",
      "Epoch 22/50\n",
      "31519/31519 [==============================] - ETA: 0s - loss: 1.4729\n",
      "Epoch 22: val_loss did not improve from 1.36033\n",
      "31519/31519 [==============================] - 20s 638us/sample - loss: 1.4729 - val_loss: 1.3877\n",
      "Epoch 23/50\n",
      "31519/31519 [==============================] - ETA: 0s - loss: 1.4229\n",
      "Epoch 23: val_loss did not improve from 1.36033\n",
      "31519/31519 [==============================] - 22s 686us/sample - loss: 1.4229 - val_loss: 1.3797\n",
      "Epoch 24/50\n",
      "31519/31519 [==============================] - ETA: 0s - loss: 1.4100\n",
      "Epoch 24: val_loss did not improve from 1.36033\n",
      "31519/31519 [==============================] - 19s 611us/sample - loss: 1.4100 - val_loss: 1.3871\n",
      "Epoch 25/50\n",
      "31519/31519 [==============================] - ETA: 0s - loss: 1.3923\n",
      "Epoch 25: val_loss did not improve from 1.36033\n",
      "31519/31519 [==============================] - 19s 614us/sample - loss: 1.3923 - val_loss: 1.3784\n",
      "Epoch 26/50\n",
      "31519/31519 [==============================] - ETA: 0s - loss: 1.3805\n",
      "Epoch 26: val_loss did not improve from 1.36033\n",
      "31519/31519 [==============================] - 19s 615us/sample - loss: 1.3805 - val_loss: 1.3828\n",
      "Epoch 27/50\n",
      "31519/31519 [==============================] - ETA: 0s - loss: 1.4299\n",
      "Epoch 27: val_loss did not improve from 1.36033\n",
      "31519/31519 [==============================] - 20s 622us/sample - loss: 1.4299 - val_loss: 1.3873\n",
      "Epoch 28/50\n",
      "31519/31519 [==============================] - ETA: 0s - loss: 1.3902\n",
      "Epoch 28: val_loss did not improve from 1.36033\n",
      "31519/31519 [==============================] - 22s 698us/sample - loss: 1.3902 - val_loss: 1.3772\n",
      "Epoch 29/50\n",
      "31519/31519 [==============================] - ETA: 0s - loss: 1.3843\n",
      "Epoch 29: val_loss did not improve from 1.36033\n",
      "31519/31519 [==============================] - 22s 695us/sample - loss: 1.3843 - val_loss: 1.3838\n",
      "Epoch 30/50\n",
      "31519/31519 [==============================] - ETA: 0s - loss: 1.3762\n",
      "Epoch 30: val_loss did not improve from 1.36033\n",
      "31519/31519 [==============================] - 22s 698us/sample - loss: 1.3762 - val_loss: 1.3722\n",
      "Epoch 31/50\n",
      "31519/31519 [==============================] - ETA: 0s - loss: 1.3751\n",
      "Epoch 31: val_loss did not improve from 1.36033\n",
      "31519/31519 [==============================] - 22s 697us/sample - loss: 1.3751 - val_loss: 1.3765\n",
      "Epoch 32/50\n",
      "31519/31519 [==============================] - ETA: 0s - loss: 1.3731\n",
      "Epoch 32: val_loss did not improve from 1.36033\n",
      "31519/31519 [==============================] - 22s 698us/sample - loss: 1.3731 - val_loss: 1.3701\n",
      "Epoch 33/50\n",
      "31519/31519 [==============================] - ETA: 0s - loss: 1.3694\n",
      "Epoch 33: val_loss did not improve from 1.36033\n",
      "31519/31519 [==============================] - 22s 687us/sample - loss: 1.3694 - val_loss: 1.3740\n",
      "Epoch 34/50\n",
      "31519/31519 [==============================] - ETA: 0s - loss: 1.3669\n",
      "Epoch 34: val_loss did not improve from 1.36033\n",
      "31519/31519 [==============================] - 22s 703us/sample - loss: 1.3669 - val_loss: 1.3706\n",
      "Epoch 35/50\n",
      "31519/31519 [==============================] - ETA: 0s - loss: 1.3663\n",
      "Epoch 35: val_loss did not improve from 1.36033\n",
      "31519/31519 [==============================] - 22s 701us/sample - loss: 1.3663 - val_loss: 1.3689\n",
      "Epoch 36/50\n",
      "31519/31519 [==============================] - ETA: 0s - loss: 1.3656\n",
      "Epoch 36: val_loss did not improve from 1.36033\n",
      "31519/31519 [==============================] - 22s 704us/sample - loss: 1.3656 - val_loss: 1.3697\n",
      "Epoch 37/50\n",
      "31519/31519 [==============================] - ETA: 0s - loss: 1.3650\n",
      "Epoch 37: val_loss did not improve from 1.36033\n",
      "31519/31519 [==============================] - 22s 704us/sample - loss: 1.3650 - val_loss: 1.3705\n",
      "Epoch 38/50\n",
      "31519/31519 [==============================] - ETA: 0s - loss: 1.3615\n",
      "Epoch 38: val_loss did not improve from 1.36033\n",
      "31519/31519 [==============================] - 20s 637us/sample - loss: 1.3615 - val_loss: 1.3705\n",
      "Epoch 39/50\n",
      "31519/31519 [==============================] - ETA: 0s - loss: 1.3613\n",
      "Epoch 39: val_loss did not improve from 1.36033\n",
      "31519/31519 [==============================] - 20s 626us/sample - loss: 1.3613 - val_loss: 1.3646\n",
      "Epoch 40/50\n",
      "31519/31519 [==============================] - ETA: 0s - loss: 1.3626\n",
      "Epoch 40: val_loss did not improve from 1.36033\n",
      "31519/31519 [==============================] - 19s 617us/sample - loss: 1.3626 - val_loss: 1.3679\n",
      "Epoch 41/50\n",
      "31519/31519 [==============================] - ETA: 0s - loss: 1.3577\n",
      "Epoch 41: val_loss did not improve from 1.36033\n",
      "31519/31519 [==============================] - 20s 637us/sample - loss: 1.3577 - val_loss: 1.3667\n",
      "Epoch 42/50\n",
      "31519/31519 [==============================] - ETA: 0s - loss: 1.3574\n",
      "Epoch 42: val_loss did not improve from 1.36033\n",
      "31519/31519 [==============================] - 20s 629us/sample - loss: 1.3574 - val_loss: 1.3623\n",
      "Epoch 43/50\n",
      "31519/31519 [==============================] - ETA: 0s - loss: 1.3565\n",
      "Epoch 43: val_loss improved from 1.36033 to 1.35835, saving model to ./checkpoints/unknown_person_baseline_p21_21.h5\n",
      "31519/31519 [==============================] - 21s 656us/sample - loss: 1.3565 - val_loss: 1.3583\n",
      "Epoch 44/50\n",
      "31519/31519 [==============================] - ETA: 0s - loss: 1.3533\n",
      "Epoch 44: val_loss did not improve from 1.35835\n",
      "31519/31519 [==============================] - 19s 606us/sample - loss: 1.3533 - val_loss: 1.3635\n",
      "Epoch 45/50\n",
      "31519/31519 [==============================] - ETA: 0s - loss: 1.3563\n",
      "Epoch 45: val_loss did not improve from 1.35835\n",
      "31519/31519 [==============================] - 20s 636us/sample - loss: 1.3563 - val_loss: 1.3664\n",
      "Epoch 46/50\n",
      "31519/31519 [==============================] - ETA: 0s - loss: 1.3552\n",
      "Epoch 46: val_loss did not improve from 1.35835\n",
      "31519/31519 [==============================] - 20s 633us/sample - loss: 1.3552 - val_loss: 1.3659\n",
      "Epoch 47/50\n",
      "31519/31519 [==============================] - ETA: 0s - loss: 1.3525\n",
      "Epoch 47: val_loss did not improve from 1.35835\n",
      "31519/31519 [==============================] - 21s 668us/sample - loss: 1.3525 - val_loss: 1.3612\n",
      "Epoch 48/50\n",
      "31519/31519 [==============================] - ETA: 0s - loss: 1.3476\n",
      "Epoch 48: val_loss did not improve from 1.35835\n",
      "31519/31519 [==============================] - 22s 694us/sample - loss: 1.3476 - val_loss: 1.3642\n",
      "Epoch 49/50\n",
      "31519/31519 [==============================] - ETA: 0s - loss: 1.3505\n",
      "Epoch 49: val_loss did not improve from 1.35835\n",
      "31519/31519 [==============================] - 22s 700us/sample - loss: 1.3505 - val_loss: 1.3660\n",
      "Epoch 50/50\n",
      "31519/31519 [==============================] - ETA: 0s - loss: 1.3487\n",
      "Epoch 50: val_loss did not improve from 1.35835\n",
      "31519/31519 [==============================] - 22s 696us/sample - loss: 1.3487 - val_loss: 1.3602\n",
      "WARNING:tensorflow:Layer lstm will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_1 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_2 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_3 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_4 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_5 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_6 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_7 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_8 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_9 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_10 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_11 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_12 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_13 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_14 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_15 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_16 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_17 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_18 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_19 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_20 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_21 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_22 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_23 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_24 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_25 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_26 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_27 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_28 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_29 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_30 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_31 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_32 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_33 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_34 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_35 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_36 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "2023-11-08 12:02:54.037360: W tensorflow/c/c_api.cc:304] Operation '{name:'lstm_13_3/lstm_cell_124/bias/Assign' id:59507 op device:{requested: '', assigned: ''} def:{{{node lstm_13_3/lstm_cell_124/bias/Assign}} = AssignVariableOp[_has_manual_control_dependencies=true, dtype=DT_FLOAT, validate_shape=false](lstm_13_3/lstm_cell_124/bias, lstm_13_3/lstm_cell_124/bias/Initializer/concat)}}' was changed by setting attribute after it was run by a session. This mutation will have no effect, and will trigger an error in the future. Either don't modify nodes after running them or create a new session.\n",
      "2023-11-08 12:02:57.343121: W tensorflow/c/c_api.cc:304] Operation '{name:'lstm_24_3/lstm_cell_135/recurrent_kernel/m/Assign' id:64375 op device:{requested: '', assigned: ''} def:{{{node lstm_24_3/lstm_cell_135/recurrent_kernel/m/Assign}} = AssignVariableOp[_has_manual_control_dependencies=true, dtype=DT_FLOAT, validate_shape=false](lstm_24_3/lstm_cell_135/recurrent_kernel/m, lstm_24_3/lstm_cell_135/recurrent_kernel/m/Initializer/zeros)}}' was changed by setting attribute after it was run by a session. This mutation will have no effect, and will trigger an error in the future. Either don't modify nodes after running them or create a new session.\n",
      "2023-11-08 12:03:00.473325: W tensorflow/c/c_api.cc:304] Operation '{name:'dropout_4_3/cond/Merge' id:63349 op device:{requested: '', assigned: ''} def:{{{node dropout_4_3/cond/Merge}} = Merge[N=2, T=DT_FLOAT, _has_manual_control_dependencies=true](dropout_4_3/cond/Identity, dropout_4_3/cond/dropout/SelectV2)}}' was changed by setting attribute after it was run by a session. This mutation will have no effect, and will trigger an error in the future. Either don't modify nodes after running them or create a new session.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(1485, 1968)\n",
      "(1514, 1968)\n",
      "(1644, 1968)\n",
      "(1764, 1968)\n",
      "(1836, 1968)\n",
      "(1699, 1968)\n",
      "(1369, 1968)\n",
      "(1766, 1968)\n",
      "(1619, 1968)\n",
      "(1692, 1968)\n",
      "(1550, 1968)\n",
      "(1920, 1968)\n",
      "(1739, 1968)\n",
      "(1872, 1968)\n",
      "(1692, 1968)\n",
      "(1824, 1968)\n",
      "(970, 1968)\n",
      "(1668, 1968)\n",
      "(1896, 1968)\n",
      "{1: 8.737275310242595, 2: 6.4926850406786, 4: 9.304172508161688, 5: 6.368476570028207, 6: 6.089857360160695, 8: 9.897217692222718, 9: 6.925582114334561, 10: 8.547458014835822, 11: 6.822249322638502, 12: 10.0, 13: 8.048801840641058, 17: 9.813930108051803, 19: 8.218550200767567, 22: 1.0, 25: 7.9571541774162915, 26: 7.309462391464195, 27: 4.797479303905, 28: 6.50530346085127, 29: 2.5542810626413615}\n",
      "Train on 31519 samples, validate on 3514 samples\n",
      "Epoch 1/20\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "2023-11-08 12:07:09.084957: E tensorflow/core/grappler/optimizers/meta_optimizer.cc:954] layout failed: INVALID_ARGUMENT: Size of values 0 does not match size of permutation 4 @ fanin shape indropout_7/cond/dropout/SelectV2-2-TransposeNHWCToNCHW-LayoutOptimizer\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "31519/31519 [==============================] - ETA: 0s - loss: 10.0916\n",
      "Epoch 1: val_loss improved from inf to 1.36280, saving model to ./checkpoints/unknown_person_p21_21.h5\n",
      "31519/31519 [==============================] - 30s 961us/sample - loss: 10.0916 - val_loss: 1.3628\n",
      "Epoch 2/20\n",
      "31519/31519 [==============================] - ETA: 0s - loss: 10.0452\n",
      "Epoch 2: val_loss did not improve from 1.36280\n",
      "31519/31519 [==============================] - 20s 620us/sample - loss: 10.0452 - val_loss: 1.3653\n",
      "Epoch 3/20\n",
      "31519/31519 [==============================] - ETA: 0s - loss: 9.9686\n",
      "Epoch 3: val_loss did not improve from 1.36280\n",
      "31519/31519 [==============================] - 21s 674us/sample - loss: 9.9686 - val_loss: 1.3660\n",
      "Epoch 4/20\n",
      "31519/31519 [==============================] - ETA: 0s - loss: 9.9296\n",
      "Epoch 4: val_loss did not improve from 1.36280\n",
      "31519/31519 [==============================] - 22s 702us/sample - loss: 9.9296 - val_loss: 1.3659\n",
      "Epoch 5/20\n",
      "31519/31519 [==============================] - ETA: 0s - loss: 9.8749\n",
      "Epoch 5: val_loss improved from 1.36280 to 1.36064, saving model to ./checkpoints/unknown_person_p21_21.h5\n",
      "31519/31519 [==============================] - 22s 704us/sample - loss: 9.8749 - val_loss: 1.3606\n",
      "Epoch 6/20\n",
      "31519/31519 [==============================] - ETA: 0s - loss: 9.8713\n",
      "Epoch 6: val_loss did not improve from 1.36064\n",
      "31519/31519 [==============================] - 22s 701us/sample - loss: 9.8713 - val_loss: 1.3654\n",
      "Epoch 7/20\n",
      "31519/31519 [==============================] - ETA: 0s - loss: 9.8512\n",
      "Epoch 7: val_loss improved from 1.36064 to 1.35749, saving model to ./checkpoints/unknown_person_p21_21.h5\n",
      "31519/31519 [==============================] - 21s 680us/sample - loss: 9.8512 - val_loss: 1.3575\n",
      "Epoch 8/20\n",
      "31519/31519 [==============================] - ETA: 0s - loss: 9.8253\n",
      "Epoch 8: val_loss did not improve from 1.35749\n",
      "31519/31519 [==============================] - 20s 642us/sample - loss: 9.8253 - val_loss: 1.3699\n",
      "Epoch 9/20\n",
      "31519/31519 [==============================] - ETA: 0s - loss: 9.8069\n",
      "Epoch 9: val_loss improved from 1.35749 to 1.35449, saving model to ./checkpoints/unknown_person_p21_21.h5\n",
      "31519/31519 [==============================] - 21s 663us/sample - loss: 9.8069 - val_loss: 1.3545\n",
      "Epoch 10/20\n",
      "31519/31519 [==============================] - ETA: 0s - loss: 9.7815\n",
      "Epoch 10: val_loss did not improve from 1.35449\n",
      "31519/31519 [==============================] - 22s 699us/sample - loss: 9.7815 - val_loss: 1.3586\n",
      "Epoch 11/20\n",
      "31519/31519 [==============================] - ETA: 0s - loss: 9.7865\n",
      "Epoch 11: val_loss improved from 1.35449 to 1.35281, saving model to ./checkpoints/unknown_person_p21_21.h5\n",
      "31519/31519 [==============================] - 22s 705us/sample - loss: 9.7865 - val_loss: 1.3528\n",
      "Epoch 12/20\n",
      "31519/31519 [==============================] - ETA: 0s - loss: 9.7978\n",
      "Epoch 12: val_loss improved from 1.35281 to 1.35092, saving model to ./checkpoints/unknown_person_p21_21.h5\n",
      "31519/31519 [==============================] - 22s 703us/sample - loss: 9.7978 - val_loss: 1.3509\n",
      "Epoch 13/20\n",
      "31519/31519 [==============================] - ETA: 0s - loss: 9.7350\n",
      "Epoch 13: val_loss did not improve from 1.35092\n",
      "31519/31519 [==============================] - 21s 675us/sample - loss: 9.7350 - val_loss: 1.3534\n",
      "Epoch 14/20\n",
      "31519/31519 [==============================] - ETA: 0s - loss: 9.7285\n",
      "Epoch 14: val_loss did not improve from 1.35092\n",
      "31519/31519 [==============================] - 21s 681us/sample - loss: 9.7285 - val_loss: 1.3586\n",
      "Epoch 15/20\n",
      "31519/31519 [==============================] - ETA: 0s - loss: 9.7392\n",
      "Epoch 15: val_loss did not improve from 1.35092\n",
      "31519/31519 [==============================] - 22s 696us/sample - loss: 9.7392 - val_loss: 1.3578\n",
      "Epoch 16/20\n",
      "31519/31519 [==============================] - ETA: 0s - loss: 9.7043\n",
      "Epoch 16: val_loss improved from 1.35092 to 1.35065, saving model to ./checkpoints/unknown_person_p21_21.h5\n",
      "31519/31519 [==============================] - 20s 645us/sample - loss: 9.7043 - val_loss: 1.3507\n",
      "Epoch 17/20\n",
      "31519/31519 [==============================] - ETA: 0s - loss: 9.7045\n",
      "Epoch 17: val_loss did not improve from 1.35065\n",
      "31519/31519 [==============================] - 19s 606us/sample - loss: 9.7045 - val_loss: 1.3550\n",
      "Epoch 18/20\n",
      "31519/31519 [==============================] - ETA: 0s - loss: 9.7209\n",
      "Epoch 18: val_loss did not improve from 1.35065\n",
      "31519/31519 [==============================] - 19s 610us/sample - loss: 9.7209 - val_loss: 1.3621\n",
      "Epoch 19/20\n",
      "31519/31519 [==============================] - ETA: 0s - loss: 9.7055\n",
      "Epoch 19: val_loss did not improve from 1.35065\n",
      "31519/31519 [==============================] - 19s 603us/sample - loss: 9.7055 - val_loss: 1.3551\n",
      "Epoch 20/20\n",
      "31519/31519 [==============================] - ETA: 0s - loss: 9.6809\n",
      "Epoch 20: val_loss improved from 1.35065 to 1.34930, saving model to ./checkpoints/unknown_person_p21_21.h5\n",
      "31519/31519 [==============================] - 19s 608us/sample - loss: 9.6809 - val_loss: 1.3493\n",
      "WARNING:tensorflow:Layer lstm will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_1 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_2 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_3 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_4 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_5 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_6 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_7 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_8 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_9 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_10 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_11 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_12 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_13 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_14 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_15 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_16 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_17 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_18 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_19 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_20 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_21 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_22 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_23 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_24 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_25 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_26 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_27 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_28 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_29 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_30 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_31 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_32 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_33 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_34 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_35 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_36 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "2023-11-08 12:14:18.956291: W tensorflow/c/c_api.cc:304] Operation '{name:'lstm_16_4/lstm_cell_164/kernel/Assign' id:79356 op device:{requested: '', assigned: ''} def:{{{node lstm_16_4/lstm_cell_164/kernel/Assign}} = AssignVariableOp[_has_manual_control_dependencies=true, dtype=DT_FLOAT, validate_shape=false](lstm_16_4/lstm_cell_164/kernel, lstm_16_4/lstm_cell_164/kernel/Initializer/random_uniform)}}' was changed by setting attribute after it was run by a session. This mutation will have no effect, and will trigger an error in the future. Either don't modify nodes after running them or create a new session.\n",
      "2023-11-08 12:14:23.161301: W tensorflow/c/c_api.cc:304] Operation '{name:'lstm_26_4/lstm_cell_174/recurrent_kernel/v/Assign' id:84446 op device:{requested: '', assigned: ''} def:{{{node lstm_26_4/lstm_cell_174/recurrent_kernel/v/Assign}} = AssignVariableOp[_has_manual_control_dependencies=true, dtype=DT_FLOAT, validate_shape=false](lstm_26_4/lstm_cell_174/recurrent_kernel/v, lstm_26_4/lstm_cell_174/recurrent_kernel/v/Initializer/zeros)}}' was changed by setting attribute after it was run by a session. This mutation will have no effect, and will trigger an error in the future. Either don't modify nodes after running them or create a new session.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Train on 31519 samples, validate on 3514 samples\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "2023-11-08 12:14:29.846629: W tensorflow/c/c_api.cc:304] Operation '{name:'loss_9/mul' id:83337 op device:{requested: '', assigned: ''} def:{{{node loss_9/mul}} = Mul[T=DT_FLOAT, _has_manual_control_dependencies=true](loss_9/mul/x, loss_9/dense_3_loss/value)}}' was changed by setting attribute after it was run by a session. This mutation will have no effect, and will trigger an error in the future. Either don't modify nodes after running them or create a new session.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Epoch 1/20\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "2023-11-08 12:14:51.312439: E tensorflow/core/grappler/optimizers/meta_optimizer.cc:954] layout failed: INVALID_ARGUMENT: Size of values 0 does not match size of permutation 4 @ fanin shape indropout_8/cond/dropout/SelectV2-2-TransposeNHWCToNCHW-LayoutOptimizer\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "31519/31519 [==============================] - ETA: 0s - loss: 1.3582"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "2023-11-08 12:15:11.817837: W tensorflow/c/c_api.cc:304] Operation '{name:'loss_9/mul' id:83337 op device:{requested: '', assigned: ''} def:{{{node loss_9/mul}} = Mul[T=DT_FLOAT, _has_manual_control_dependencies=true](loss_9/mul/x, loss_9/dense_3_loss/value)}}' was changed by setting attribute after it was run by a session. This mutation will have no effect, and will trigger an error in the future. Either don't modify nodes after running them or create a new session.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Epoch 1: val_loss improved from inf to 1.36607, saving model to ./checkpoints/unknown_person_baseline_2_p21_21.h5\n",
      "31519/31519 [==============================] - 30s 940us/sample - loss: 1.3582 - val_loss: 1.3661\n",
      "Epoch 2/20\n",
      "31519/31519 [==============================] - ETA: 0s - loss: 1.3557\n",
      "Epoch 2: val_loss did not improve from 1.36607\n",
      "31519/31519 [==============================] - 19s 608us/sample - loss: 1.3557 - val_loss: 1.3706\n",
      "Epoch 3/20\n",
      "31519/31519 [==============================] - ETA: 0s - loss: 1.3504\n",
      "Epoch 3: val_loss improved from 1.36607 to 1.35891, saving model to ./checkpoints/unknown_person_baseline_2_p21_21.h5\n",
      "31519/31519 [==============================] - 19s 611us/sample - loss: 1.3504 - val_loss: 1.3589\n",
      "Epoch 4/20\n",
      "31519/31519 [==============================] - ETA: 0s - loss: 1.3489\n",
      "Epoch 4: val_loss did not improve from 1.35891\n",
      "31519/31519 [==============================] - 19s 597us/sample - loss: 1.3489 - val_loss: 1.3600\n",
      "Epoch 5/20\n",
      "31519/31519 [==============================] - ETA: 0s - loss: 1.3472\n",
      "Epoch 5: val_loss improved from 1.35891 to 1.35865, saving model to ./checkpoints/unknown_person_baseline_2_p21_21.h5\n",
      "31519/31519 [==============================] - 19s 601us/sample - loss: 1.3472 - val_loss: 1.3587\n",
      "Epoch 6/20\n",
      "31519/31519 [==============================] - ETA: 0s - loss: 1.3446\n",
      "Epoch 6: val_loss improved from 1.35865 to 1.35817, saving model to ./checkpoints/unknown_person_baseline_2_p21_21.h5\n",
      "31519/31519 [==============================] - 20s 631us/sample - loss: 1.3446 - val_loss: 1.3582\n",
      "Epoch 7/20\n",
      "31519/31519 [==============================] - ETA: 0s - loss: 1.3438\n",
      "Epoch 7: val_loss did not improve from 1.35817\n",
      "31519/31519 [==============================] - 22s 694us/sample - loss: 1.3438 - val_loss: 1.3615\n",
      "Epoch 8/20\n",
      "31519/31519 [==============================] - ETA: 0s - loss: 1.3425\n",
      "Epoch 8: val_loss did not improve from 1.35817\n",
      "31519/31519 [==============================] - 21s 680us/sample - loss: 1.3425 - val_loss: 1.3619\n",
      "Epoch 9/20\n",
      "31519/31519 [==============================] - ETA: 0s - loss: 1.3447\n",
      "Epoch 9: val_loss improved from 1.35817 to 1.35778, saving model to ./checkpoints/unknown_person_baseline_2_p21_21.h5\n",
      "31519/31519 [==============================] - 20s 642us/sample - loss: 1.3447 - val_loss: 1.3578\n",
      "Epoch 10/20\n",
      "31519/31519 [==============================] - ETA: 0s - loss: 1.3394\n",
      "Epoch 10: val_loss improved from 1.35778 to 1.35723, saving model to ./checkpoints/unknown_person_baseline_2_p21_21.h5\n",
      "31519/31519 [==============================] - 20s 628us/sample - loss: 1.3394 - val_loss: 1.3572\n",
      "Epoch 11/20\n",
      "31519/31519 [==============================] - ETA: 0s - loss: 1.3397\n",
      "Epoch 11: val_loss improved from 1.35723 to 1.35452, saving model to ./checkpoints/unknown_person_baseline_2_p21_21.h5\n",
      "31519/31519 [==============================] - 20s 623us/sample - loss: 1.3397 - val_loss: 1.3545\n",
      "Epoch 12/20\n",
      "31519/31519 [==============================] - ETA: 0s - loss: 1.3416\n",
      "Epoch 12: val_loss improved from 1.35452 to 1.35140, saving model to ./checkpoints/unknown_person_baseline_2_p21_21.h5\n",
      "31519/31519 [==============================] - 20s 623us/sample - loss: 1.3416 - val_loss: 1.3514\n",
      "Epoch 13/20\n",
      "31519/31519 [==============================] - ETA: 0s - loss: 1.3380\n",
      "Epoch 13: val_loss did not improve from 1.35140\n",
      "31519/31519 [==============================] - 22s 690us/sample - loss: 1.3380 - val_loss: 1.3578\n",
      "Epoch 14/20\n",
      "31519/31519 [==============================] - ETA: 0s - loss: 1.3369\n",
      "Epoch 14: val_loss did not improve from 1.35140\n",
      "31519/31519 [==============================] - 22s 694us/sample - loss: 1.3369 - val_loss: 1.3541\n",
      "Epoch 15/20\n",
      "31519/31519 [==============================] - ETA: 0s - loss: 1.3338\n",
      "Epoch 15: val_loss did not improve from 1.35140\n",
      "31519/31519 [==============================] - 22s 700us/sample - loss: 1.3338 - val_loss: 1.3519\n",
      "Epoch 16/20\n",
      "31519/31519 [==============================] - ETA: 0s - loss: 1.3337\n",
      "Epoch 16: val_loss improved from 1.35140 to 1.34838, saving model to ./checkpoints/unknown_person_baseline_2_p21_21.h5\n",
      "31519/31519 [==============================] - 22s 702us/sample - loss: 1.3337 - val_loss: 1.3484\n",
      "Epoch 17/20\n",
      "31519/31519 [==============================] - ETA: 0s - loss: 1.3351\n",
      "Epoch 17: val_loss improved from 1.34838 to 1.34819, saving model to ./checkpoints/unknown_person_baseline_2_p21_21.h5\n",
      "31519/31519 [==============================] - 19s 612us/sample - loss: 1.3351 - val_loss: 1.3482\n",
      "Epoch 18/20\n",
      "31519/31519 [==============================] - ETA: 0s - loss: 1.3329\n",
      "Epoch 18: val_loss did not improve from 1.34819\n",
      "31519/31519 [==============================] - 20s 649us/sample - loss: 1.3329 - val_loss: 1.3542\n",
      "Epoch 19/20\n",
      "31519/31519 [==============================] - ETA: 0s - loss: 1.3319\n",
      "Epoch 19: val_loss did not improve from 1.34819\n",
      "31519/31519 [==============================] - 19s 616us/sample - loss: 1.3319 - val_loss: 1.3512\n",
      "Epoch 20/20\n",
      "31519/31519 [==============================] - ETA: 0s - loss: 1.3298\n",
      "Epoch 20: val_loss improved from 1.34819 to 1.34796, saving model to ./checkpoints/unknown_person_baseline_2_p21_21.h5\n",
      "31519/31519 [==============================] - 20s 644us/sample - loss: 1.3298 - val_loss: 1.3480\n",
      "Train on 31519 samples, validate on 3514 samples\n",
      "Epoch 1/50\n",
      "31519/31519 [==============================] - ETA: 0s - loss: 1.3274\n",
      "Epoch 1: val_loss improved from inf to 1.35751, saving model to ./checkpoints/unknown_person_baseline_p21_22.h5\n",
      "31519/31519 [==============================] - 22s 704us/sample - loss: 1.3274 - val_loss: 1.3575\n",
      "Epoch 2/50\n",
      "31519/31519 [==============================] - ETA: 0s - loss: 1.3289\n",
      "Epoch 2: val_loss improved from 1.35751 to 1.34443, saving model to ./checkpoints/unknown_person_baseline_p21_22.h5\n",
      "31519/31519 [==============================] - 22s 703us/sample - loss: 1.3289 - val_loss: 1.3444\n",
      "Epoch 3/50\n",
      "31519/31519 [==============================] - ETA: 0s - loss: 1.3278\n",
      "Epoch 3: val_loss did not improve from 1.34443\n",
      "31519/31519 [==============================] - 22s 695us/sample - loss: 1.3278 - val_loss: 1.3449\n",
      "Epoch 4/50\n",
      "31519/31519 [==============================] - ETA: 0s - loss: 1.3223\n",
      "Epoch 4: val_loss did not improve from 1.34443\n",
      "31519/31519 [==============================] - 20s 623us/sample - loss: 1.3223 - val_loss: 1.3477\n",
      "Epoch 5/50\n",
      "31519/31519 [==============================] - ETA: 0s - loss: 1.3231\n",
      "Epoch 5: val_loss did not improve from 1.34443\n",
      "31519/31519 [==============================] - 20s 619us/sample - loss: 1.3231 - val_loss: 1.3488\n",
      "Epoch 6/50\n",
      "31519/31519 [==============================] - ETA: 0s - loss: 1.3223\n",
      "Epoch 6: val_loss improved from 1.34443 to 1.34428, saving model to ./checkpoints/unknown_person_baseline_p21_22.h5\n",
      "31519/31519 [==============================] - 20s 622us/sample - loss: 1.3223 - val_loss: 1.3443\n",
      "Epoch 7/50\n",
      "31519/31519 [==============================] - ETA: 0s - loss: 1.3233\n",
      "Epoch 7: val_loss did not improve from 1.34428\n",
      "31519/31519 [==============================] - 20s 640us/sample - loss: 1.3233 - val_loss: 1.3452\n",
      "Epoch 8/50\n",
      "31519/31519 [==============================] - ETA: 0s - loss: 1.3391\n",
      "Epoch 8: val_loss did not improve from 1.34428\n",
      "31519/31519 [==============================] - 20s 650us/sample - loss: 1.3391 - val_loss: 1.3451\n",
      "Epoch 9/50\n",
      "31519/31519 [==============================] - ETA: 0s - loss: 1.3279\n",
      "Epoch 9: val_loss improved from 1.34428 to 1.34386, saving model to ./checkpoints/unknown_person_baseline_p21_22.h5\n",
      "31519/31519 [==============================] - 19s 614us/sample - loss: 1.3279 - val_loss: 1.3439\n",
      "Epoch 10/50\n",
      "31519/31519 [==============================] - ETA: 0s - loss: 1.3303\n",
      "Epoch 10: val_loss did not improve from 1.34386\n",
      "31519/31519 [==============================] - 21s 660us/sample - loss: 1.3303 - val_loss: 1.3452\n",
      "Epoch 11/50\n",
      "31519/31519 [==============================] - ETA: 0s - loss: 1.3289\n",
      "Epoch 11: val_loss did not improve from 1.34386\n",
      "31519/31519 [==============================] - 22s 696us/sample - loss: 1.3289 - val_loss: 1.3502\n",
      "Epoch 12/50\n",
      "31519/31519 [==============================] - ETA: 0s - loss: 1.3362\n",
      "Epoch 12: val_loss did not improve from 1.34386\n",
      "31519/31519 [==============================] - 22s 696us/sample - loss: 1.3362 - val_loss: 1.3461\n",
      "Epoch 13/50\n",
      "31519/31519 [==============================] - ETA: 0s - loss: 1.3332\n",
      "Epoch 13: val_loss did not improve from 1.34386\n",
      "31519/31519 [==============================] - 21s 659us/sample - loss: 1.3332 - val_loss: 1.3455\n",
      "Epoch 14/50\n",
      "31519/31519 [==============================] - ETA: 0s - loss: 1.3435\n",
      "Epoch 14: val_loss did not improve from 1.34386\n",
      "31519/31519 [==============================] - 22s 700us/sample - loss: 1.3435 - val_loss: 1.3511\n",
      "Epoch 15/50\n",
      "31519/31519 [==============================] - ETA: 0s - loss: 1.3419\n",
      "Epoch 15: val_loss did not improve from 1.34386\n",
      "31519/31519 [==============================] - 22s 696us/sample - loss: 1.3419 - val_loss: 1.3550\n",
      "Epoch 16/50\n",
      "31519/31519 [==============================] - ETA: 0s - loss: 1.3463\n",
      "Epoch 16: val_loss did not improve from 1.34386\n",
      "31519/31519 [==============================] - 20s 640us/sample - loss: 1.3463 - val_loss: 1.3544\n",
      "Epoch 17/50\n",
      "31519/31519 [==============================] - ETA: 0s - loss: 1.3455\n",
      "Epoch 17: val_loss did not improve from 1.34386\n",
      "31519/31519 [==============================] - 19s 612us/sample - loss: 1.3455 - val_loss: 1.3567\n",
      "Epoch 18/50\n",
      "31519/31519 [==============================] - ETA: 0s - loss: 1.3443\n",
      "Epoch 18: val_loss did not improve from 1.34386\n",
      "31519/31519 [==============================] - 20s 622us/sample - loss: 1.3443 - val_loss: 1.3509\n",
      "Epoch 19/50\n",
      "31519/31519 [==============================] - ETA: 0s - loss: 1.3486\n",
      "Epoch 19: val_loss did not improve from 1.34386\n",
      "31519/31519 [==============================] - 21s 663us/sample - loss: 1.3486 - val_loss: 1.3549\n",
      "Epoch 20/50\n",
      "31519/31519 [==============================] - ETA: 0s - loss: 1.3544\n",
      "Epoch 20: val_loss did not improve from 1.34386\n",
      "31519/31519 [==============================] - 20s 647us/sample - loss: 1.3544 - val_loss: 1.3562\n",
      "Epoch 21/50\n",
      "31519/31519 [==============================] - ETA: 0s - loss: 1.3829\n",
      "Epoch 21: val_loss did not improve from 1.34386\n",
      "31519/31519 [==============================] - 20s 623us/sample - loss: 1.3829 - val_loss: 1.3614\n",
      "Epoch 22/50\n",
      "31519/31519 [==============================] - ETA: 0s - loss: 1.3779\n",
      "Epoch 22: val_loss did not improve from 1.34386\n",
      "31519/31519 [==============================] - 22s 691us/sample - loss: 1.3779 - val_loss: 1.3690\n",
      "Epoch 23/50\n",
      "31519/31519 [==============================] - ETA: 0s - loss: 1.3795\n",
      "Epoch 23: val_loss did not improve from 1.34386\n",
      "31519/31519 [==============================] - 22s 706us/sample - loss: 1.3795 - val_loss: 1.3690\n",
      "Epoch 24/50\n",
      "31519/31519 [==============================] - ETA: 0s - loss: 1.3744\n",
      "Epoch 24: val_loss did not improve from 1.34386\n",
      "31519/31519 [==============================] - 21s 671us/sample - loss: 1.3744 - val_loss: 1.3623\n",
      "Epoch 25/50\n",
      "31519/31519 [==============================] - ETA: 0s - loss: 1.3677\n",
      "Epoch 25: val_loss did not improve from 1.34386\n",
      "31519/31519 [==============================] - 20s 626us/sample - loss: 1.3677 - val_loss: 1.3636\n",
      "Epoch 26/50\n",
      "31519/31519 [==============================] - ETA: 0s - loss: 1.3724\n",
      "Epoch 26: val_loss did not improve from 1.34386\n",
      "31519/31519 [==============================] - 20s 619us/sample - loss: 1.3724 - val_loss: 1.3648\n",
      "Epoch 27/50\n",
      "31519/31519 [==============================] - ETA: 0s - loss: 1.3744\n",
      "Epoch 27: val_loss did not improve from 1.34386\n",
      "31519/31519 [==============================] - 19s 617us/sample - loss: 1.3744 - val_loss: 1.3731\n",
      "Epoch 28/50\n",
      "31519/31519 [==============================] - ETA: 0s - loss: 1.3728\n",
      "Epoch 28: val_loss did not improve from 1.34386\n",
      "31519/31519 [==============================] - 19s 617us/sample - loss: 1.3728 - val_loss: 1.3727\n",
      "Epoch 29/50\n",
      "31519/31519 [==============================] - ETA: 0s - loss: 1.3687\n",
      "Epoch 29: val_loss did not improve from 1.34386\n",
      "31519/31519 [==============================] - 22s 690us/sample - loss: 1.3687 - val_loss: 1.3717\n",
      "Epoch 30/50\n",
      "31519/31519 [==============================] - ETA: 0s - loss: 1.3675\n",
      "Epoch 30: val_loss did not improve from 1.34386\n",
      "31519/31519 [==============================] - 20s 627us/sample - loss: 1.3675 - val_loss: 1.3730\n",
      "Epoch 31/50\n",
      "31519/31519 [==============================] - ETA: 0s - loss: 1.3633\n",
      "Epoch 31: val_loss did not improve from 1.34386\n",
      "31519/31519 [==============================] - 20s 619us/sample - loss: 1.3633 - val_loss: 1.3679\n",
      "Epoch 32/50\n",
      "31519/31519 [==============================] - ETA: 0s - loss: 1.3668\n",
      "Epoch 32: val_loss did not improve from 1.34386\n",
      "31519/31519 [==============================] - 19s 619us/sample - loss: 1.3668 - val_loss: 1.3751\n",
      "Epoch 33/50\n",
      "31519/31519 [==============================] - ETA: 0s - loss: 1.3662\n",
      "Epoch 33: val_loss did not improve from 1.34386\n",
      "31519/31519 [==============================] - 19s 610us/sample - loss: 1.3662 - val_loss: 1.3827\n",
      "Epoch 34/50\n",
      "31519/31519 [==============================] - ETA: 0s - loss: 1.3613\n",
      "Epoch 34: val_loss did not improve from 1.34386\n",
      "31519/31519 [==============================] - 19s 614us/sample - loss: 1.3613 - val_loss: 1.3661\n",
      "Epoch 35/50\n",
      "31519/31519 [==============================] - ETA: 0s - loss: 1.3661\n",
      "Epoch 35: val_loss did not improve from 1.34386\n",
      "31519/31519 [==============================] - 19s 612us/sample - loss: 1.3661 - val_loss: 1.3724\n",
      "Epoch 36/50\n",
      "31519/31519 [==============================] - ETA: 0s - loss: 1.3623\n",
      "Epoch 36: val_loss did not improve from 1.34386\n",
      "31519/31519 [==============================] - 19s 611us/sample - loss: 1.3623 - val_loss: 1.3643\n",
      "Epoch 37/50\n",
      "31519/31519 [==============================] - ETA: 0s - loss: 1.3595\n",
      "Epoch 37: val_loss did not improve from 1.34386\n",
      "31519/31519 [==============================] - 20s 641us/sample - loss: 1.3595 - val_loss: 1.3724\n",
      "Epoch 38/50\n",
      "31519/31519 [==============================] - ETA: 0s - loss: 1.3626\n",
      "Epoch 38: val_loss did not improve from 1.34386\n",
      "31519/31519 [==============================] - 22s 697us/sample - loss: 1.3626 - val_loss: 1.3719\n",
      "Epoch 39/50\n",
      "31519/31519 [==============================] - ETA: 0s - loss: 1.3645\n",
      "Epoch 39: val_loss did not improve from 1.34386\n",
      "31519/31519 [==============================] - 22s 695us/sample - loss: 1.3645 - val_loss: 1.3679\n",
      "Epoch 40/50\n",
      "31519/31519 [==============================] - ETA: 0s - loss: 1.3656\n",
      "Epoch 40: val_loss did not improve from 1.34386\n",
      "31519/31519 [==============================] - 22s 699us/sample - loss: 1.3656 - val_loss: 1.3764\n",
      "Epoch 41/50\n",
      "31519/31519 [==============================] - ETA: 0s - loss: 1.3643\n",
      "Epoch 41: val_loss did not improve from 1.34386\n",
      "31519/31519 [==============================] - 22s 698us/sample - loss: 1.3643 - val_loss: 1.3749\n",
      "Epoch 42/50\n",
      "31519/31519 [==============================] - ETA: 0s - loss: 1.3619\n",
      "Epoch 42: val_loss did not improve from 1.34386\n",
      "31519/31519 [==============================] - 22s 695us/sample - loss: 1.3619 - val_loss: 1.3746\n",
      "Epoch 43/50\n",
      "31519/31519 [==============================] - ETA: 0s - loss: 1.3604\n",
      "Epoch 43: val_loss did not improve from 1.34386\n",
      "31519/31519 [==============================] - 20s 637us/sample - loss: 1.3604 - val_loss: 1.3759\n",
      "Epoch 44/50\n",
      "31519/31519 [==============================] - ETA: 0s - loss: 1.3577\n",
      "Epoch 44: val_loss did not improve from 1.34386\n",
      "31519/31519 [==============================] - 20s 644us/sample - loss: 1.3577 - val_loss: 1.3679\n",
      "Epoch 45/50\n",
      "31519/31519 [==============================] - ETA: 0s - loss: 1.3538\n",
      "Epoch 45: val_loss did not improve from 1.34386\n",
      "31519/31519 [==============================] - 20s 640us/sample - loss: 1.3538 - val_loss: 1.3736\n",
      "Epoch 46/50\n",
      "31519/31519 [==============================] - ETA: 0s - loss: 1.3572\n",
      "Epoch 46: val_loss did not improve from 1.34386\n",
      "31519/31519 [==============================] - 19s 607us/sample - loss: 1.3572 - val_loss: 1.3673\n",
      "Epoch 47/50\n",
      "31519/31519 [==============================] - ETA: 0s - loss: 1.3574\n",
      "Epoch 47: val_loss did not improve from 1.34386\n",
      "31519/31519 [==============================] - 19s 608us/sample - loss: 1.3574 - val_loss: 1.3700\n",
      "Epoch 48/50\n",
      "31519/31519 [==============================] - ETA: 0s - loss: 1.3534\n",
      "Epoch 48: val_loss did not improve from 1.34386\n",
      "31519/31519 [==============================] - 19s 609us/sample - loss: 1.3534 - val_loss: 1.3695\n",
      "Epoch 49/50\n",
      "31519/31519 [==============================] - ETA: 0s - loss: 1.3547\n",
      "Epoch 49: val_loss did not improve from 1.34386\n",
      "31519/31519 [==============================] - 19s 616us/sample - loss: 1.3547 - val_loss: 1.3683\n",
      "Epoch 50/50\n",
      "31519/31519 [==============================] - ETA: 0s - loss: 1.3536\n",
      "Epoch 50: val_loss did not improve from 1.34386\n",
      "31519/31519 [==============================] - 19s 613us/sample - loss: 1.3536 - val_loss: 1.3634\n",
      "WARNING:tensorflow:Layer lstm will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_1 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_2 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_3 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_4 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_5 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_6 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_7 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_8 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_9 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_10 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_11 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_12 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_13 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_14 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_15 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_16 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_17 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_18 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_19 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_20 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_21 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_22 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_23 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_24 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_25 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_26 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_27 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_28 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_29 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_30 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_31 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_32 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_33 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_34 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_35 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_36 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "2023-11-08 12:38:52.115370: W tensorflow/c/c_api.cc:304] Operation '{name:'lstm_28_5/lstm_cell_213/kernel/Assign' id:100638 op device:{requested: '', assigned: ''} def:{{{node lstm_28_5/lstm_cell_213/kernel/Assign}} = AssignVariableOp[_has_manual_control_dependencies=true, dtype=DT_FLOAT, validate_shape=false](lstm_28_5/lstm_cell_213/kernel, lstm_28_5/lstm_cell_213/kernel/Initializer/random_uniform)}}' was changed by setting attribute after it was run by a session. This mutation will have no effect, and will trigger an error in the future. Either don't modify nodes after running them or create a new session.\n",
      "2023-11-08 12:38:57.186650: W tensorflow/c/c_api.cc:304] Operation '{name:'lstm_26_5/lstm_cell_211/kernel/m/Assign' id:103158 op device:{requested: '', assigned: ''} def:{{{node lstm_26_5/lstm_cell_211/kernel/m/Assign}} = AssignVariableOp[_has_manual_control_dependencies=true, dtype=DT_FLOAT, validate_shape=false](lstm_26_5/lstm_cell_211/kernel/m, lstm_26_5/lstm_cell_211/kernel/m/Initializer/zeros)}}' was changed by setting attribute after it was run by a session. This mutation will have no effect, and will trigger an error in the future. Either don't modify nodes after running them or create a new session.\n",
      "2023-11-08 12:39:02.150456: W tensorflow/c/c_api.cc:304] Operation '{name:'dropout_4_5/cond/Merge' id:102107 op device:{requested: '', assigned: ''} def:{{{node dropout_4_5/cond/Merge}} = Merge[N=2, T=DT_FLOAT, _has_manual_control_dependencies=true](dropout_4_5/cond/Identity, dropout_4_5/cond/dropout/SelectV2)}}' was changed by setting attribute after it was run by a session. This mutation will have no effect, and will trigger an error in the future. Either don't modify nodes after running them or create a new session.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(1485, 1968)\n",
      "(1514, 1968)\n",
      "(1644, 1968)\n",
      "(1764, 1968)\n",
      "(1836, 1968)\n",
      "(1699, 1968)\n",
      "(1369, 1968)\n",
      "(1766, 1968)\n",
      "(1619, 1968)\n",
      "(1692, 1968)\n",
      "(1550, 1968)\n",
      "(1920, 1968)\n",
      "(1739, 1968)\n",
      "(1872, 1968)\n",
      "(1692, 1968)\n",
      "(1824, 1968)\n",
      "(970, 1968)\n",
      "(1668, 1968)\n",
      "(1896, 1968)\n",
      "{1: 8.673526998530303, 2: 6.030673494783035, 4: 9.481426998014493, 5: 6.376880749187622, 6: 5.811897888102374, 8: 9.904409568255359, 9: 6.8993304697136795, 10: 8.47226992458226, 11: 6.717676886368801, 12: 10.0, 13: 7.639602906018624, 17: 9.806416548339577, 19: 8.015853100239145, 22: 1.0, 25: 7.767593257296923, 26: 7.400981106607679, 27: 4.618808437527685, 28: 6.111535653820256, 29: 2.242301786249649}\n",
      "Train on 31519 samples, validate on 3514 samples\n",
      "Epoch 1/20\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "2023-11-08 12:43:01.806811: E tensorflow/core/grappler/optimizers/meta_optimizer.cc:954] layout failed: INVALID_ARGUMENT: Size of values 0 does not match size of permutation 4 @ fanin shape indropout_9/cond/dropout/SelectV2-2-TransposeNHWCToNCHW-LayoutOptimizer\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "31519/31519 [==============================] - ETA: 0s - loss: 9.7562\n",
      "Epoch 1: val_loss improved from inf to 1.36312, saving model to ./checkpoints/unknown_person_p21_22.h5\n",
      "31519/31519 [==============================] - 31s 988us/sample - loss: 9.7562 - val_loss: 1.3631\n",
      "Epoch 2/20\n",
      "31519/31519 [==============================] - ETA: 0s - loss: 9.7123\n",
      "Epoch 2: val_loss improved from 1.36312 to 1.35468, saving model to ./checkpoints/unknown_person_p21_22.h5\n",
      "31519/31519 [==============================] - 19s 604us/sample - loss: 9.7123 - val_loss: 1.3547\n",
      "Epoch 3/20\n",
      "31519/31519 [==============================] - ETA: 0s - loss: 9.7844\n",
      "Epoch 3: val_loss did not improve from 1.35468\n",
      "31519/31519 [==============================] - 20s 619us/sample - loss: 9.7844 - val_loss: 1.3607\n",
      "Epoch 4/20\n",
      "31519/31519 [==============================] - ETA: 0s - loss: 9.6635\n",
      "Epoch 4: val_loss improved from 1.35468 to 1.35216, saving model to ./checkpoints/unknown_person_p21_22.h5\n",
      "31519/31519 [==============================] - 22s 704us/sample - loss: 9.6635 - val_loss: 1.3522\n",
      "Epoch 5/20\n",
      "31519/31519 [==============================] - ETA: 0s - loss: 9.6063\n",
      "Epoch 5: val_loss did not improve from 1.35216\n",
      "31519/31519 [==============================] - 22s 701us/sample - loss: 9.6063 - val_loss: 1.3592\n",
      "Epoch 6/20\n",
      "31519/31519 [==============================] - ETA: 0s - loss: 9.5870\n",
      "Epoch 6: val_loss improved from 1.35216 to 1.35136, saving model to ./checkpoints/unknown_person_p21_22.h5\n",
      "31519/31519 [==============================] - 22s 692us/sample - loss: 9.5870 - val_loss: 1.3514\n",
      "Epoch 7/20\n",
      "31519/31519 [==============================] - ETA: 0s - loss: 9.5519\n",
      "Epoch 7: val_loss improved from 1.35136 to 1.34340, saving model to ./checkpoints/unknown_person_p21_22.h5\n",
      "31519/31519 [==============================] - 22s 686us/sample - loss: 9.5519 - val_loss: 1.3434\n",
      "Epoch 8/20\n",
      "31519/31519 [==============================] - ETA: 0s - loss: 9.5230\n",
      "Epoch 8: val_loss did not improve from 1.34340\n",
      "31519/31519 [==============================] - 20s 633us/sample - loss: 9.5230 - val_loss: 1.3453\n",
      "Epoch 9/20\n",
      "31519/31519 [==============================] - ETA: 0s - loss: 9.5372\n",
      "Epoch 9: val_loss did not improve from 1.34340\n",
      "31519/31519 [==============================] - 21s 655us/sample - loss: 9.5372 - val_loss: 1.3454\n",
      "Epoch 10/20\n",
      "31519/31519 [==============================] - ETA: 0s - loss: 9.5280\n",
      "Epoch 10: val_loss improved from 1.34340 to 1.33988, saving model to ./checkpoints/unknown_person_p21_22.h5\n",
      "31519/31519 [==============================] - 19s 615us/sample - loss: 9.5280 - val_loss: 1.3399\n",
      "Epoch 11/20\n",
      "31519/31519 [==============================] - ETA: 0s - loss: 9.4813\n",
      "Epoch 11: val_loss did not improve from 1.33988\n",
      "31519/31519 [==============================] - 19s 614us/sample - loss: 9.4813 - val_loss: 1.3417\n",
      "Epoch 12/20\n",
      "31519/31519 [==============================] - ETA: 0s - loss: 9.4912\n",
      "Epoch 12: val_loss did not improve from 1.33988\n",
      "31519/31519 [==============================] - 19s 614us/sample - loss: 9.4912 - val_loss: 1.3423\n",
      "Epoch 13/20\n",
      "31519/31519 [==============================] - ETA: 0s - loss: 9.4838\n",
      "Epoch 13: val_loss did not improve from 1.33988\n",
      "31519/31519 [==============================] - 21s 660us/sample - loss: 9.4838 - val_loss: 1.3489\n",
      "Epoch 14/20\n",
      "31519/31519 [==============================] - ETA: 0s - loss: 9.4544\n",
      "Epoch 14: val_loss did not improve from 1.33988\n",
      "31519/31519 [==============================] - 22s 697us/sample - loss: 9.4544 - val_loss: 1.3472\n",
      "Epoch 15/20\n",
      "31519/31519 [==============================] - ETA: 0s - loss: 9.4474\n",
      "Epoch 15: val_loss did not improve from 1.33988\n",
      "31519/31519 [==============================] - 22s 702us/sample - loss: 9.4474 - val_loss: 1.3507\n",
      "Epoch 16/20\n",
      "31519/31519 [==============================] - ETA: 0s - loss: 9.4560\n",
      "Epoch 16: val_loss did not improve from 1.33988\n",
      "31519/31519 [==============================] - 22s 695us/sample - loss: 9.4560 - val_loss: 1.3476\n",
      "Epoch 17/20\n",
      "31519/31519 [==============================] - ETA: 0s - loss: 9.4492\n",
      "Epoch 17: val_loss did not improve from 1.33988\n",
      "31519/31519 [==============================] - 22s 697us/sample - loss: 9.4492 - val_loss: 1.3469\n",
      "Epoch 18/20\n",
      "31519/31519 [==============================] - ETA: 0s - loss: 9.4470\n",
      "Epoch 18: val_loss did not improve from 1.33988\n",
      "31519/31519 [==============================] - 21s 653us/sample - loss: 9.4470 - val_loss: 1.3556\n",
      "Epoch 19/20\n",
      "31519/31519 [==============================] - ETA: 0s - loss: 9.4282\n",
      "Epoch 19: val_loss did not improve from 1.33988\n",
      "31519/31519 [==============================] - 19s 618us/sample - loss: 9.4282 - val_loss: 1.3489\n",
      "Epoch 20/20\n",
      "31519/31519 [==============================] - ETA: 0s - loss: 9.4433\n",
      "Epoch 20: val_loss did not improve from 1.33988\n",
      "31519/31519 [==============================] - 19s 616us/sample - loss: 9.4433 - val_loss: 1.3474\n",
      "WARNING:tensorflow:Layer lstm will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_1 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_2 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_3 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_4 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_5 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_6 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_7 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_8 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_9 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_10 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_11 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_12 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_13 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_14 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_15 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_16 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_17 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_18 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_19 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_20 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_21 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_22 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_23 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_24 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_25 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_26 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_27 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_28 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_29 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_30 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_31 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_32 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_33 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_34 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_35 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_36 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "2023-11-08 12:50:09.187276: W tensorflow/c/c_api.cc:304] Operation '{name:'lstm_22_6/lstm_cell_244/recurrent_kernel/Assign' id:119096 op device:{requested: '', assigned: ''} def:{{{node lstm_22_6/lstm_cell_244/recurrent_kernel/Assign}} = AssignVariableOp[_has_manual_control_dependencies=true, dtype=DT_FLOAT, validate_shape=false](lstm_22_6/lstm_cell_244/recurrent_kernel, lstm_22_6/lstm_cell_244/recurrent_kernel/Initializer/mul_1)}}' was changed by setting attribute after it was run by a session. This mutation will have no effect, and will trigger an error in the future. Either don't modify nodes after running them or create a new session.\n",
      "2023-11-08 12:50:15.393368: W tensorflow/c/c_api.cc:304] Operation '{name:'lstm_14_6/lstm_cell_236/kernel/m/Assign' id:122376 op device:{requested: '', assigned: ''} def:{{{node lstm_14_6/lstm_cell_236/kernel/m/Assign}} = AssignVariableOp[_has_manual_control_dependencies=true, dtype=DT_FLOAT, validate_shape=false](lstm_14_6/lstm_cell_236/kernel/m, lstm_14_6/lstm_cell_236/kernel/m/Initializer/zeros)}}' was changed by setting attribute after it was run by a session. This mutation will have no effect, and will trigger an error in the future. Either don't modify nodes after running them or create a new session.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Train on 31519 samples, validate on 3514 samples\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "2023-11-08 12:50:24.570968: W tensorflow/c/c_api.cc:304] Operation '{name:'loss_13/mul' id:122095 op device:{requested: '', assigned: ''} def:{{{node loss_13/mul}} = Mul[T=DT_FLOAT, _has_manual_control_dependencies=true](loss_13/mul/x, loss_13/dense_3_loss/value)}}' was changed by setting attribute after it was run by a session. This mutation will have no effect, and will trigger an error in the future. Either don't modify nodes after running them or create a new session.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Epoch 1/20\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "2023-11-08 12:50:57.255517: E tensorflow/core/grappler/optimizers/meta_optimizer.cc:954] layout failed: INVALID_ARGUMENT: Size of values 0 does not match size of permutation 4 @ fanin shape indropout_10/cond/dropout/SelectV2-2-TransposeNHWCToNCHW-LayoutOptimizer\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "31519/31519 [==============================] - ETA: 0s - loss: 1.3255"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "2023-11-08 12:51:18.643664: W tensorflow/c/c_api.cc:304] Operation '{name:'loss_13/mul' id:122095 op device:{requested: '', assigned: ''} def:{{{node loss_13/mul}} = Mul[T=DT_FLOAT, _has_manual_control_dependencies=true](loss_13/mul/x, loss_13/dense_3_loss/value)}}' was changed by setting attribute after it was run by a session. This mutation will have no effect, and will trigger an error in the future. Either don't modify nodes after running them or create a new session.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Epoch 1: val_loss improved from inf to 1.34307, saving model to ./checkpoints/unknown_person_baseline_2_p21_22.h5\n",
      "31519/31519 [==============================] - 35s 1ms/sample - loss: 1.3255 - val_loss: 1.3431\n",
      "Epoch 2/20\n",
      "31519/31519 [==============================] - ETA: 0s - loss: 1.3218\n",
      "Epoch 2: val_loss did not improve from 1.34307\n",
      "31519/31519 [==============================] - 20s 625us/sample - loss: 1.3218 - val_loss: 1.3443\n",
      "Epoch 3/20\n",
      "31519/31519 [==============================] - ETA: 0s - loss: 1.3542\n",
      "Epoch 3: val_loss did not improve from 1.34307\n",
      "31519/31519 [==============================] - 19s 613us/sample - loss: 1.3542 - val_loss: 1.3541\n",
      "Epoch 4/20\n",
      "31519/31519 [==============================] - ETA: 0s - loss: 1.3295\n",
      "Epoch 4: val_loss did not improve from 1.34307\n",
      "31519/31519 [==============================] - 19s 615us/sample - loss: 1.3295 - val_loss: 1.3467\n",
      "Epoch 5/20\n",
      "31519/31519 [==============================] - ETA: 0s - loss: 1.3210\n",
      "Epoch 5: val_loss did not improve from 1.34307\n",
      "31519/31519 [==============================] - 19s 610us/sample - loss: 1.3210 - val_loss: 1.3440\n",
      "Epoch 6/20\n",
      "31519/31519 [==============================] - ETA: 0s - loss: 1.3235\n",
      "Epoch 6: val_loss did not improve from 1.34307\n",
      "31519/31519 [==============================] - 19s 612us/sample - loss: 1.3235 - val_loss: 1.3450\n",
      "Epoch 7/20\n",
      "31519/31519 [==============================] - ETA: 0s - loss: 1.3211\n",
      "Epoch 7: val_loss did not improve from 1.34307\n",
      "31519/31519 [==============================] - 19s 608us/sample - loss: 1.3211 - val_loss: 1.3449\n",
      "Epoch 8/20\n",
      "31519/31519 [==============================] - ETA: 0s - loss: 1.3215\n",
      "Epoch 8: val_loss improved from 1.34307 to 1.34268, saving model to ./checkpoints/unknown_person_baseline_2_p21_22.h5\n",
      "31519/31519 [==============================] - 21s 681us/sample - loss: 1.3215 - val_loss: 1.3427\n",
      "Epoch 9/20\n",
      "31519/31519 [==============================] - ETA: 0s - loss: 1.3183\n",
      "Epoch 9: val_loss did not improve from 1.34268\n",
      "31519/31519 [==============================] - 19s 618us/sample - loss: 1.3183 - val_loss: 1.3502\n",
      "Epoch 10/20\n",
      "31519/31519 [==============================] - ETA: 0s - loss: 1.3200\n",
      "Epoch 10: val_loss improved from 1.34268 to 1.33777, saving model to ./checkpoints/unknown_person_baseline_2_p21_22.h5\n",
      "31519/31519 [==============================] - 20s 620us/sample - loss: 1.3200 - val_loss: 1.3378\n",
      "Epoch 11/20\n",
      "31519/31519 [==============================] - ETA: 0s - loss: 1.3162\n",
      "Epoch 11: val_loss did not improve from 1.33777\n",
      "31519/31519 [==============================] - 20s 619us/sample - loss: 1.3162 - val_loss: 1.3394\n",
      "Epoch 12/20\n",
      "31519/31519 [==============================] - ETA: 0s - loss: 1.3169\n",
      "Epoch 12: val_loss did not improve from 1.33777\n",
      "31519/31519 [==============================] - 19s 610us/sample - loss: 1.3169 - val_loss: 1.3402\n",
      "Epoch 13/20\n",
      "31519/31519 [==============================] - ETA: 0s - loss: 1.3168\n",
      "Epoch 13: val_loss did not improve from 1.33777\n",
      "31519/31519 [==============================] - 22s 693us/sample - loss: 1.3168 - val_loss: 1.3411\n",
      "Epoch 14/20\n",
      "31519/31519 [==============================] - ETA: 0s - loss: 1.3148\n",
      "Epoch 14: val_loss did not improve from 1.33777\n",
      "31519/31519 [==============================] - 21s 681us/sample - loss: 1.3148 - val_loss: 1.3410\n",
      "Epoch 15/20\n",
      "31519/31519 [==============================] - ETA: 0s - loss: 1.3123\n",
      "Epoch 15: val_loss did not improve from 1.33777\n",
      "31519/31519 [==============================] - 20s 622us/sample - loss: 1.3123 - val_loss: 1.3432\n",
      "Epoch 16/20\n",
      "31519/31519 [==============================] - ETA: 0s - loss: 1.3127\n",
      "Epoch 16: val_loss did not improve from 1.33777\n",
      "31519/31519 [==============================] - 20s 619us/sample - loss: 1.3127 - val_loss: 1.3433\n",
      "Epoch 17/20\n",
      "31519/31519 [==============================] - ETA: 0s - loss: 1.3114\n",
      "Epoch 17: val_loss did not improve from 1.33777\n",
      "31519/31519 [==============================] - 19s 613us/sample - loss: 1.3114 - val_loss: 1.3398\n",
      "Epoch 18/20\n",
      "31519/31519 [==============================] - ETA: 0s - loss: 1.3106\n",
      "Epoch 18: val_loss did not improve from 1.33777\n",
      "31519/31519 [==============================] - 19s 610us/sample - loss: 1.3106 - val_loss: 1.3429\n",
      "Epoch 19/20\n",
      "31519/31519 [==============================] - ETA: 0s - loss: 1.3107\n",
      "Epoch 19: val_loss improved from 1.33777 to 1.33670, saving model to ./checkpoints/unknown_person_baseline_2_p21_22.h5\n",
      "31519/31519 [==============================] - 19s 617us/sample - loss: 1.3107 - val_loss: 1.3367\n",
      "Epoch 20/20\n",
      "31519/31519 [==============================] - ETA: 0s - loss: 1.3071\n",
      "Epoch 20: val_loss improved from 1.33670 to 1.33641, saving model to ./checkpoints/unknown_person_baseline_2_p21_22.h5\n",
      "31519/31519 [==============================] - 19s 611us/sample - loss: 1.3071 - val_loss: 1.3364\n"
     ]
    }
   ],
   "source": [
    "person_nums = [1,2,4,5,6,8,9,10,11,12,13,17,19,21,22,25,26,27,28,29]\n",
    "## Build the baseline model for emotion recognition with dropout layers\n",
    "from tensorflow.keras.models import Model\n",
    "from tensorflow.keras.layers import Input, Dense, concatenate, SimpleRNN, LSTM, Conv2D, Flatten, MaxPooling2D, GRU, AveragePooling2D, Dropout, Lambda\n",
    "from tensorflow.keras.callbacks import ModelCheckpoint\n",
    "from sklearn.preprocessing import StandardScaler\n",
    "def individual_model(features_list):\n",
    "    input_layers = []\n",
    "    hidden_layers = []\n",
    "    combined_feature = np.empty((len(features_list[0]),0))\n",
    "    for i, feature in enumerate(features_list):\n",
    "        \n",
    "        if len(feature.shape) == 3:\n",
    "            input_i = Input(shape=(feature.shape[1], feature.shape[2]))\n",
    "            input_layers.append(input_i)\n",
    "\n",
    "            hidden_i = input_i[:,:,:,None]\n",
    "            hidden_i = Conv2D(32, 3, activation='relu')(hidden_i)\n",
    "            hidden_i = AveragePooling2D((3, 3))(hidden_i)\n",
    "            hidden_i = Dropout(0.5)(hidden_i)\n",
    "            hidden_i = Conv2D(16, 3, activation='relu')(hidden_i)\n",
    "            hidden_i = AveragePooling2D((3, 3))(hidden_i)\n",
    "            hidden_i = Dropout(0.5)(hidden_i)\n",
    "            hidden_i = Conv2D(8, 1, activation='relu')(hidden_i)\n",
    "            hidden_i = AveragePooling2D((2, 2))(hidden_i)\n",
    "            hidden_i = Dropout(0.5)(hidden_i)\n",
    "            hidden_i = Conv2D(4, 1, activation='relu')(hidden_i)\n",
    "            hidden_i = AveragePooling2D((2, 2))(hidden_i)\n",
    "            hidden_i = Dropout(0.5)(hidden_i)\n",
    "            hidden_i = Flatten()(hidden_i)\n",
    "\n",
    "            hidden_layers.append(hidden_i)\n",
    "        elif feature.shape[1] <20:\n",
    "            combined_feature = np.hstack((combined_feature, feature))\n",
    "            \n",
    "        else:  # For series features\n",
    "            input_i = Input(shape=(feature.shape[1],))\n",
    "            input_layers.append(input_i)\n",
    "            hidden_i = Lambda(lambda x: x[:, :, None])(input_i)  # Add a new dimension\n",
    "            hidden_i = LSTM(4)(hidden_i)\n",
    "            hidden_layers.append(hidden_i)\n",
    "    input_i = Input(shape=(combined_feature.shape[1],))\n",
    "    input_layers.append(input_i)\n",
    "    dense_num = np.max((1, int(combined_feature.shape[1]/2)))\n",
    "    hidden_i = Dense(dense_num, activation='relu')(input_i)\n",
    "    hidden_layers.append(hidden_i)\n",
    "    print(combined_feature.shape)\n",
    "    concat_layer = concatenate(hidden_layers)\n",
    "    h = Dropout(0.2)(concat_layer)\n",
    "    h = Dense(64, activation='relu')(h)\n",
    "    h = Dense(32, activation='relu')(h)\n",
    "    output_layer = Dense(2)(h)\n",
    "    model = Model(inputs=input_layers, outputs=output_layer)\n",
    "    model.compile(loss='mean_absolute_error', optimizer='adam')\n",
    "    return model\n",
    "\n",
    "import tensorflow as tf\n",
    "from tensorflow.keras.callbacks import Callback\n",
    "import numpy as np\n",
    "\n",
    "class PruningCallback(tf.keras.callbacks.Callback):\n",
    "    def __init__(self, initial_pruning_factor=0.1, final_pruning_factor=0.5, start_epoch=0, end_epoch=None, frequency=1):\n",
    "        super(PruningCallback, self).__init__()\n",
    "        self.initial_pruning_factor = initial_pruning_factor\n",
    "        self.final_pruning_factor = final_pruning_factor\n",
    "        self.start_epoch = start_epoch\n",
    "        self.end_epoch = end_epoch if end_epoch is not None else np.inf\n",
    "        self.frequency = frequency\n",
    "        self.pruned_weights = {}\n",
    "        self.layer_importance = {}\n",
    "\n",
    "    def get_pruning_factor(self, epoch):\n",
    "        if epoch < self.start_epoch:\n",
    "            return 0\n",
    "        if epoch > self.end_epoch:\n",
    "            return self.final_pruning_factor\n",
    "        return self.initial_pruning_factor + (self.final_pruning_factor - self.initial_pruning_factor) * (epoch - self.start_epoch) / (self.end_epoch - self.start_epoch)\n",
    "\n",
    "    def on_train_begin(self, logs=None):\n",
    "        total_weight_magnitude = 0\n",
    "        for layer in self.model.layers:\n",
    "            if hasattr(layer, 'get_weights'):\n",
    "                weights = layer.get_weights()\n",
    "                layer_norm = sum(np.linalg.norm(w) for w in weights)\n",
    "                total_weight_magnitude += layer_norm\n",
    "                self.layer_importance[layer.name] = layer_norm\n",
    "    \n",
    "        # Normalize the layer importance values so they sum up to 1\n",
    "        for layer_name in self.layer_importance:\n",
    "            self.layer_importance[layer_name] /= total_weight_magnitude\n",
    "    # def on_train_begin(self, logs=None):\n",
    "    #     total_weight_magnitude = sum([np.linalg.norm(layer.get_weights()) for layer in self.model.layers if hasattr(layer, 'get_weights')])\n",
    "    #     for layer in self.model.layers:\n",
    "    #         if hasattr(layer, 'get_weights'):\n",
    "    #             self.layer_importance[layer.name] = np.linalg.norm(layer.get_weights()) / total_weight_magnitude\n",
    "\n",
    "    def get_layer_pruning_factor(self, layer_name, global_pruning_factor):\n",
    "        if layer_name in self.layer_importance:\n",
    "            importance = self.layer_importance[layer_name]\n",
    "            adjusted_pruning_factor = global_pruning_factor * (1 - importance)\n",
    "            return min(adjusted_pruning_factor, 1)  # Ensure the pruning factor is not greater than 1\n",
    "        return global_pruning_factor\n",
    "    def prune_weights(self, layer, global_pruning_factor):\n",
    "        \n",
    "        weights = layer.get_weights()\n",
    "        layer_name = layer.name\n",
    "        pruning_factor = self.get_layer_pruning_factor(layer_name, global_pruning_factor)\n",
    "\n",
    "        if layer_name not in self.pruned_weights:\n",
    "            self.pruned_weights[layer_name] = [np.zeros_like(w, dtype=bool) for w in weights]\n",
    "        \n",
    "        for i in range(len(weights)):\n",
    "            weight = weights[i]\n",
    "            # print(weight.shape)\n",
    "            # print(weight.size)\n",
    "            if weight.ndim > 1:  # Only prune dense or convolutional layers\n",
    "                unpruned_weights = np.logical_not(self.pruned_weights[layer_name][i])\n",
    "                num_unpruned = np.sum(unpruned_weights)\n",
    "                num_pruning = min(num_unpruned, int(weight.size * pruning_factor) - np.sum(self.pruned_weights[layer_name][i]))\n",
    "                num_pruning = int(weight.size * pruning_factor) - np.sum(self.pruned_weights[layer_name][i])\n",
    "                if num_pruning > 0:\n",
    "                    unpruned_flat_indices = np.flatnonzero(unpruned_weights)\n",
    "                    abs_unpruned_weights = np.abs(weight[unpruned_weights])\n",
    "                    pruning_flat_indices = np.argpartition(abs_unpruned_weights, num_pruning)[:num_pruning]\n",
    "                    \n",
    "                    indices = np.unravel_index(pruning_flat_indices, weight.shape)\n",
    "                    self.pruned_weights[layer_name][i][indices] = True\n",
    "\n",
    "                weights[i] = weights[i]*(~self.pruned_weights[layer_name][i])\n",
    "                \n",
    "        layer.set_weights(weights)\n",
    "\n",
    "    def on_epoch_end(self, epoch, logs=None):\n",
    "        if (epoch - self.start_epoch) % self.frequency != 0:\n",
    "            return\n",
    "\n",
    "        pruning_factor = self.get_pruning_factor(epoch)\n",
    "        for layer in self.model.layers:\n",
    "            if isinstance(layer, (tf.keras.layers.Dense, tf.keras.layers.Conv2D)):\n",
    "                self.prune_weights(layer, pruning_factor)\n",
    "from feature_emotion import feature_extract, split_data, label_unique_tuples, split_data_unknown\n",
    "gts, sensor_nums, walk_nums, trace_nums, people_nums, spe_centr, delta_spe_centr, spe_crest, delta_spe_crest, spe_decrease, delta_spe_decrease, spe_entropy, delta_spe_entropy, spe_flatness, delta_spe_flatness, spe_flux, delta_spe_flux, spe_kurtosis, delta_spe_kurtosis, spe_skewness, delta_spe_skewness, spe_rfp, delta_spe_rfp, spe_slope, delta_spe_slope, spe_spread, delta_spe_spread, wlk_fre, wlk_fres_trace, cwt_figs_all, cwt_sum_all_0, cwt_sum_all_1, cwt_sum_all_2, cwt_sum_all_3, high_fre_compos, pitchs, low_fre_compos, auto_corrs, real_hils, imag_hils, dur_time_1_alls, dur_time_2_alls, jitters, shimmers, jitter_rap, hrs, feature, slope, zcrs, fft_features, energy_alls, log_energy_alls, smoothe_energy_alls, legendres, double_support_time, pdps_new, lpcs, ceps_features = feature_extract(person_nums)\n",
    "\n",
    "walk_nums_all = np.squeeze(walk_nums)\n",
    "trace_nums_all = np.squeeze(trace_nums)\n",
    "people_nums_all = np.squeeze(people_nums)\n",
    "\n",
    "## 0: train, 1: validation 2: test\n",
    "\n",
    "test_person_id = [21]\n",
    "flag_tr_val_te = split_data_unknown(test_person_id, walk_nums_all, trace_nums_all, people_nums_all, person_nums, rand_seed = 42)\n",
    "## Data Normalization before training ans testing\n",
    "import tensorflow as tf\n",
    "tf.compat.v1.disable_v2_behavior()\n",
    "from tensorflow.keras.models import Model\n",
    "from tensorflow.keras.layers import Input, Dense, concatenate, LSTM\n",
    "from sklearn.preprocessing import StandardScaler\n",
    "scalers = []\n",
    "X_train_normalized = []\n",
    "X_val_normalized = []\n",
    "X_test_normalized = []\n",
    "train_idx = np.where(flag_tr_val_te ==0)[0]\n",
    "np.random.shuffle(train_idx)\n",
    "val_idx = np.where(flag_tr_val_te ==1)[0]\n",
    "test_idx = np.where(flag_tr_val_te ==2)[0]\n",
    "\n",
    "for i, feature in enumerate([spe_centr, delta_spe_centr, spe_crest, delta_spe_crest, spe_decrease, delta_spe_decrease, spe_entropy, delta_spe_entropy, spe_flatness, delta_spe_flatness, spe_flux, delta_spe_flux, spe_kurtosis, delta_spe_kurtosis, spe_skewness, delta_spe_skewness, spe_rfp, delta_spe_rfp, spe_slope, delta_spe_slope, spe_spread, delta_spe_spread, wlk_fre, wlk_fres_trace, cwt_figs_all, cwt_sum_all_0, cwt_sum_all_1, cwt_sum_all_2, cwt_sum_all_3, high_fre_compos, pitchs, low_fre_compos, auto_corrs, real_hils, imag_hils, dur_time_1_alls, dur_time_2_alls, jitters, shimmers, jitter_rap, hrs, feature, slope, zcrs, fft_features, energy_alls, log_energy_alls, smoothe_energy_alls, legendres, double_support_time, pdps_new, lpcs, ceps_features]):\n",
    "    scaler = StandardScaler()\n",
    "    if len(feature.shape)==2:\n",
    "        X_train_i = feature[train_idx,:]\n",
    "        X_val_i = feature[val_idx,:]\n",
    "        X_test_i = feature[test_idx,:]\n",
    "        X_train_normalized_i = scaler.fit_transform(X_train_i)\n",
    "        X_val_normalized_i = scaler.transform(X_val_i)\n",
    "        X_test_normalized_i = scaler.transform(X_test_i)\n",
    "        scalers.append(scaler)\n",
    "    else:\n",
    "        X_train_i = feature[train_idx,:,:]\n",
    "        X_val_i = feature[val_idx,:,:]\n",
    "        X_test_i = feature[test_idx,:,:]\n",
    "        X_train_normalized_i = scaler.fit_transform(X_train_i.reshape(X_train_i.shape[0], -1)).reshape(X_train_i.shape)\n",
    "        X_val_normalized_i = scaler.transform(X_val_i.reshape(X_val_i.shape[0], -1)).reshape(X_val_i.shape)\n",
    "        X_test_normalized_i = scaler.transform(X_test_i.reshape(X_test_i.shape[0], -1)).reshape(X_test_i.shape)\n",
    "        scalers.append(scaler)\n",
    "    X_train_normalized.append(X_train_normalized_i)\n",
    "    X_val_normalized.append(X_val_normalized_i)\n",
    "    X_test_normalized.append(X_test_normalized_i)\n",
    "y_train = gts[train_idx,:]\n",
    "y_val = gts[val_idx,:]\n",
    "y_test = gts[test_idx,:]\n",
    "X_train_normalized_new = []\n",
    "combined_feature = np.empty((len(X_train_normalized[0]),0))\n",
    "for feature in X_train_normalized:\n",
    "    if len(feature.shape) == 3:\n",
    "        X_train_normalized_new.append(feature)\n",
    "    elif feature.shape[1] <20:\n",
    "        combined_feature = np.hstack((combined_feature, feature))\n",
    "    else:\n",
    "        X_train_normalized_new.append(feature)\n",
    "X_train_normalized_new.append(combined_feature)\n",
    "\n",
    "X_val_normalized_new = []\n",
    "combined_feature = np.empty((len(X_val_normalized[0]),0))\n",
    "for feature in X_val_normalized:\n",
    "    if len(feature.shape) == 3:\n",
    "        X_val_normalized_new.append(feature)\n",
    "    elif feature.shape[1] <20:\n",
    "        combined_feature = np.hstack((combined_feature, feature))\n",
    "    else:\n",
    "        X_val_normalized_new.append(feature)\n",
    "X_val_normalized_new.append(combined_feature)\n",
    "\n",
    "X_test_normalized_new = []\n",
    "combined_feature = np.empty((len(X_test_normalized[0]),0))\n",
    "for feature in X_test_normalized:\n",
    "    if len(feature.shape) == 3:\n",
    "        X_test_normalized_new.append(feature)\n",
    "    elif feature.shape[1] <20:\n",
    "        combined_feature = np.hstack((combined_feature, feature))\n",
    "    else:\n",
    "        X_test_normalized_new.append(feature)\n",
    "X_test_normalized_new.append(combined_feature)\n",
    "\n",
    "model = individual_model(X_train_normalized)\n",
    "model.compile(loss='mean_absolute_error', optimizer='adam')\n",
    "num_epochs = 50\n",
    "\n",
    "\n",
    "# Gradually prune weights in dense and convolutional layers from 10% to 50% over the course of training, starting from epoch 0.\n",
    "\n",
    "\n",
    "rates = [0.4, 0.5, 0.6]\n",
    "idx = 20\n",
    "for r in rates:\n",
    "    model_name = './checkpoints/unknown_person_baseline_p'+str(test_person_id[0])+'_'+str(idx)+'.h5'\n",
    "    checkpoint = ModelCheckpoint(model_name, monitor='val_loss', mode='min', save_best_only=True, verbose=1)\n",
    "\n",
    "    pruning_callback = PruningCallback(initial_pruning_factor=0.01, final_pruning_factor=r, start_epoch=5, end_epoch=20, frequency=1)\n",
    "    model.fit(X_train_normalized_new, y_train, epochs=num_epochs, batch_size=1024, validation_data=(X_val_normalized_new, y_val), callbacks=[pruning_callback, checkpoint])\n",
    "    import tensorflow as tf\n",
    "    model = tf.keras.models.load_model(model_name)\n",
    "    from tensorflow.keras.models import Model\n",
    "    layers = model.layers\n",
    "    second_last_layer_output = layers[-4].output\n",
    "    feature_extractor_model = Model(inputs=model.input, outputs=second_last_layer_output)\n",
    "    train_features = feature_extractor_model.predict(X_train_normalized_new)\n",
    "    test_features = feature_extractor_model.predict(X_test_normalized_new)\n",
    "    \n",
    "    p_train = people_nums[train_idx,:]\n",
    "    p_val = people_nums[val_idx,:]\n",
    "    p_test = people_nums[test_idx,:]\n",
    "    ## Calculate the distance between test person and training person\n",
    "    def euclidean_distance(a, b):\n",
    "        return np.sqrt(np.sum((a - b) ** 2, axis=1))\n",
    "    \n",
    "    distance_dict = {}\n",
    "    for ii in range(len(person_nums)):\n",
    "        if person_nums[ii] == test_person_id[0]:\n",
    "            continue\n",
    "        else:\n",
    "            \n",
    "            ind = np.where(p_train ==person_nums[ii])[0]\n",
    "            tmp_train_features = train_features[ind, :]\n",
    "            distances = np.array([euclidean_distance(train_sample, test_features) for train_sample in tmp_train_features])\n",
    "            print(distances.shape)\n",
    "            average_distances = np.mean(distances, axis=1)\n",
    "    \n",
    "            # Step 4: Find the overall average distance\n",
    "            overall_average_distance = np.mean(average_distances)\n",
    "            distance_dict[person_nums[ii]] = overall_average_distance\n",
    "    \n",
    "    \n",
    "    def normalize_to_weights(distance_dict):\n",
    "        distances = np.array(list(distance_dict.values()))\n",
    "        # Handle the case where a distance is zero to avoid division by zero\n",
    "        distances = np.clip(distances, a_min=1e-10, a_max=None)\n",
    "        weights = 1 / distances\n",
    "        normalized_weights = weights\n",
    "        # normalized_weights = weights / sum(weights)\n",
    "        # print(sum(weights))\n",
    "        # print(sum(normalized_weights))\n",
    "        # Assign the normalized weights back to the dictionary\n",
    "        normalized_weight_dict = dict(zip(distance_dict.keys(), normalized_weights))\n",
    "        return normalized_weight_dict\n",
    "    def scale_dict_values(my_dict):\n",
    "        scaled_dict = my_dict.copy()\n",
    "        min_val = min(scaled_dict.values())\n",
    "        max_val = max(scaled_dict.values())\n",
    "        \n",
    "        for key in scaled_dict:\n",
    "            scaled_dict[key] = 1 + 9 * (scaled_dict[key] - min_val) / (max_val - min_val)\n",
    "        \n",
    "        return scaled_dict\n",
    "    weights_dict = normalize_to_weights(distance_dict)\n",
    "    weights_dict = scale_dict_values(weights_dict)\n",
    "    print(weights_dict)\n",
    "    \n",
    "    w_train = np.zeros_like(p_train)\n",
    "    for i in range(len(w_train)):\n",
    "        w_train[i] = weights_dict[int(p_train[i])]\n",
    "    \n",
    "    w_train = np.squeeze(w_train)\n",
    "    \n",
    "    import sys\n",
    "    sys.path.append('..')\n",
    "    # for layer in model.layers[-4:]:\n",
    "    #     layer.trainable = False\n",
    "    # model = individual_model.individual_model(X_train_normalized)\n",
    "    model_name = './checkpoints/unknown_person_p'+str(test_person_id[0])+'_'+str(idx)+'.h5'\n",
    "    checkpoint = ModelCheckpoint(model_name, monitor='val_loss', mode='min', save_best_only=True, verbose=1)\n",
    "    pruning_callback = PruningCallback(initial_pruning_factor=r, final_pruning_factor=r, start_epoch=1, end_epoch=20, frequency=1)\n",
    "    \n",
    "    history = model.fit(x=X_train_normalized_new, y=y_train,sample_weight= w_train,epochs=20, batch_size=1024, validation_data=(X_val_normalized_new, y_val), callbacks=[pruning_callback, checkpoint])\n",
    "    import sys\n",
    "    sys.path.append('..')\n",
    "    # for layer in model.layers[-4:]:\n",
    "    #     layer.trainable = False\n",
    "    # model = individual_model.individual_model(X_train_normalized)\n",
    "    model_name = './checkpoints/unknown_person_baseline_p'+str(test_person_id[0])+'_'+str(idx)+'.h5'\n",
    "    model = tf.keras.models.load_model(model_name)\n",
    "    model_name = './checkpoints/unknown_person_baseline_2_p'+str(test_person_id[0])+'_'+str(idx)+'.h5'\n",
    "    idx += 1\n",
    "    checkpoint = ModelCheckpoint(model_name, monitor='val_loss', mode='min', save_best_only=True, verbose=1)\n",
    "    pruning_callback = PruningCallback(initial_pruning_factor=r, final_pruning_factor=r, start_epoch=1, end_epoch=20, frequency=1)\n",
    "    \n",
    "    history = model.fit(x=X_train_normalized_new, y=y_train,epochs=20, batch_size=1024, validation_data=(X_val_normalized_new, y_val), callbacks=[pruning_callback, checkpoint])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "572086b2-c248-4bb1-ac76-9af500e2b4f7",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "f71b59e4-6581-4ea8-994d-a02a6654ab6b",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.11.5"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
